diff -ur root-5.34.23.orig/net/netxng/src/TNetXNGFile.cxx root-5.34.23/net/netxng/src/TNetXNGFile.cxx
--- root-5.34.23.orig/net/netxng/src/TNetXNGFile.cxx	2014-11-07 15:06:59.000000000 +0100
+++ root-5.34.23/net/netxng/src/TNetXNGFile.cxx	2014-12-19 10:11:33.343919333 +0100
@@ -28,6 +28,7 @@
 #include <XrdCl/XrdClFile.hh>
 #include <XrdCl/XrdClXRootDResponses.hh>
 #include <XrdCl/XrdClDefaultEnv.hh>
+#include <XrdVersion.hh>
 #include <iostream>
 
 //------------------------------------------------------------------------------
