diff -ur root-5.32.00.orig/hist/hist/src/TGraph2DErrors.cxx root-5.32.00/hist/hist/src/TGraph2DErrors.cxx
--- root-5.32.00.orig/hist/hist/src/TGraph2DErrors.cxx	2011-12-02 12:42:07.000000000 +0100
+++ root-5.32.00/hist/hist/src/TGraph2DErrors.cxx	2012-02-11 15:11:23.205768374 +0100
@@ -71,7 +71,7 @@
    dte->SetLineColor(kBlue-3);
    dte->SetLineWidth(2);
    dte->Draw("err p0");
-   gPad->SetLogy(1);
+   TVirtualPad::Pad()->SetLogy(1);
    return c;
 }
 End_Macro
diff -ur root-5.32.00.orig/hist/hist/src/TMultiGraph.cxx root-5.32.00/hist/hist/src/TMultiGraph.cxx
--- root-5.32.00.orig/hist/hist/src/TMultiGraph.cxx	2011-12-02 12:42:07.000000000 +0100
+++ root-5.32.00/hist/hist/src/TMultiGraph.cxx	2012-02-11 16:23:09.975601997 +0100
@@ -157,7 +157,7 @@
    mg->GetYaxis()->SetTitle("Coefficients");
 
    // Change the axis limits
-   gPad->Modified();
+   TVirtualPad::Pad()->Modified();
    mg->GetXaxis()->SetLimits(1.5,7.5);
    mg->SetMinimum(0.);
    mg->SetMaximum(10.);
diff -ur root-5.32.00.orig/hist/histpainter/src/TGraphPainter.cxx root-5.32.00/hist/histpainter/src/TGraphPainter.cxx
--- root-5.32.00.orig/hist/histpainter/src/TGraphPainter.cxx	2011-12-02 12:42:07.000000000 +0100
+++ root-5.32.00/hist/histpainter/src/TGraphPainter.cxx	2012-02-11 15:11:23.242767900 +0100
@@ -334,8 +334,8 @@
    float y[]     = {1,4,9};
    TGraphErrors tg(3,x,y,err_x,err_y);
    c48->Divide(2,1);
-   c48->cd(1); gPad->DrawFrame(0,0,4,8); tg.Draw("PC");
-   c48->cd(2); gPad->DrawFrame(0,0,4,8); tg.Draw("0PC");
+   c48->cd(1); TVirtualPad::Pad()->DrawFrame(0,0,4,8); tg.Draw("PC");
+   c48->cd(2); TVirtualPad::Pad()->DrawFrame(0,0,4,8); tg.Draw("0PC");
    return c48;
 }
 End_Macro
diff -ur root-5.32.00.orig/hist/histpainter/src/THistPainter.cxx root-5.32.00/hist/histpainter/src/THistPainter.cxx
--- root-5.32.00.orig/hist/histpainter/src/THistPainter.cxx	2011-12-02 12:42:07.000000000 +0100
+++ root-5.32.00/hist/histpainter/src/THistPainter.cxx	2012-02-11 15:15:38.786479566 +0100
@@ -665,14 +665,14 @@
 
    // scale hint1 to the pad coordinates
    Float_t rightmax = 1.1*hint1->GetMaximum();
-   Float_t scale = gPad->GetUymax()/rightmax;
+   Float_t scale = TVirtualPad::Pad()->GetUymax()/rightmax;
    hint1->SetLineColor(kRed);
    hint1->Scale(scale);
    hint1->Draw("same");
 
    // draw an axis on the right side
-   TGaxis *axis = new TGaxis(gPad->GetUxmax(),gPad->GetUymin(),
-   gPad->GetUxmax(), gPad->GetUymax(),0,rightmax,510,"+L");
+   TGaxis *axis = new TGaxis(TVirtualPad::Pad()->GetUxmax(),TVirtualPad::Pad()->GetUymin(),
+   TVirtualPad::Pad()->GetUxmax(), TVirtualPad::Pad()->GetUymax(),0,rightmax,510,"+L");
    axis->SetLineColor(kRed);
    axis->SetTextColor(kRed);
    axis->Draw();
@@ -2178,7 +2178,7 @@
 
    TMultiGraph *mg;
    TKey *key;
-   TIter nextkey(gDirectory->GetListOfKeys());
+   TIter nextkey(TDirectory::CurrentDirectory()->GetListOfKeys());
    while (key = (TKey*)nextkey()) {
       obj = key->ReadObj();
       if (obj->InheritsFrom("TMultiGraph")) {
diff -ur root-5.32.00.orig/hist/histpainter/src/TPaletteAxis.cxx root-5.32.00/hist/histpainter/src/TPaletteAxis.cxx
--- root-5.32.00.orig/hist/histpainter/src/TPaletteAxis.cxx	2011-12-02 12:42:07.000000000 +0100
+++ root-5.32.00/hist/histpainter/src/TPaletteAxis.cxx	2012-02-11 16:28:01.762786179 +0100
@@ -58,7 +58,7 @@
    }
    gStyle->SetPalette(1);
    h2->Draw("COLZ");
-   gPad->Update();
+   TVirtualPad::Pad()->Update();
    TPaletteAxis *palette = (TPaletteAxis*)h2->GetListOfFunctions()->FindObject("palette");
    palette->SetY2NDC(0.7);
    return c1;
diff -ur root-5.32.00.orig/tree/treeviewer/src/TParallelCoord.cxx root-5.32.00/tree/treeviewer/src/TParallelCoord.cxx
--- root-5.32.00.orig/tree/treeviewer/src/TParallelCoord.cxx	2011-12-02 12:42:02.000000000 +0100
+++ root-5.32.00/tree/treeviewer/src/TParallelCoord.cxx	2012-02-11 16:30:34.884789287 +0100
@@ -78,7 +78,7 @@
    TFile *f = TFile::Open("$ROOTSYS/tutorials/tree/cernstaff.root");
    TTree *T = (TTree*)f->Get("T");
    T->Draw("Age:Grade:Step:Cost:Division:Nation","","para");
-   TParallelCoord* para = (TParallelCoord*)gPad->GetListOfPrimitives()->FindObject("ParaCoord");
+   TParallelCoord* para = (TParallelCoord*)TVirtualPad::Pad()->GetListOfPrimitives()->FindObject("ParaCoord");
    TParallelCoordVar* grade = (TParallelCoordVar*)para->GetVarList()->FindObject("Grade");
    grade->AddRange(new TParallelCoordRange(grade,11.5,14));
    para->AddSelection("less30");
diff -ur root-5.32.00.orig/tutorials/graphics/earth.C root-5.32.00/tutorials/graphics/earth.C
--- root-5.32.00.orig/tutorials/graphics/earth.C	2011-12-02 12:41:50.000000000 +0100
+++ root-5.32.00/tutorials/graphics/earth.C	2012-02-11 16:07:56.025452212 +0100
@@ -19,7 +19,7 @@
    TH2F *hs = new TH2F("hs","Sinusoidal",180, -180, 180, 181, -90.5, 90.5);
    TH2F *hp = new TH2F("hp","Parabolic", 180, -180, 180, 181, -90.5, 90.5);
 
-   TString dat = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
+   TString dat = gSystem->UnixPathName(TInterpreter::Instance()->GetCurrentMacroName());
    dat.ReplaceAll("C","dat");
    dat.ReplaceAll("/./","/");
 
diff -ur root-5.32.00.orig/tutorials/hist/hbars.C root-5.32.00/tutorials/hist/hbars.C
--- root-5.32.00.orig/tutorials/hist/hbars.C	2011-12-02 12:41:49.000000000 +0100
+++ root-5.32.00/tutorials/hist/hbars.C	2012-02-11 16:05:25.309365607 +0100
@@ -1,10 +1,10 @@
 // Example of bar charts with 1-d histograms
 // Author: Rene Brun
 {
-   TString dir = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
+   TString dir = gSystem->UnixPathName(TInterpreter::Instance()->GetCurrentMacroName());
    dir.ReplaceAll("hbars.C","../tree/cernbuild.C");
    dir.ReplaceAll("/./","/");
-   if (!gInterpreter->IsLoaded(dir.Data())) gInterpreter->LoadMacro(dir.Data());
+   if (!TInterpreter::Instance()->IsLoaded(dir.Data())) TInterpreter::Instance()->LoadMacro(dir.Data());
    TFile *f = (TFile*)gROOT->ProcessLineFast("cernbuild(1)");
    if (!f) return;
    TTree *T = (TTree*)f->Get("T");
@@ -14,13 +14,13 @@
    c1->Divide(1,2);
    
    //horizontal bar chart
-   c1->cd(1); gPad->SetGrid(); gPad->SetLogx(); gPad->SetFrameFillColor(33);
+   c1->cd(1); TVirtualPad::Pad()->SetGrid(); TVirtualPad::Pad()->SetLogx(); TVirtualPad::Pad()->SetFrameFillColor(33);
    T->Draw("Nation","","hbar2");
    
    //vertical bar chart
-   c1->cd(2); gPad->SetGrid(); gPad->SetFrameFillColor(33);
+   c1->cd(2); TVirtualPad::Pad()->SetGrid(); TVirtualPad::Pad()->SetFrameFillColor(33);
    T->Draw("Division>>hDiv","","goff");
-   TH1F *hDiv   = (TH1F*)gDirectory->Get("hDiv");
+   TH1F *hDiv   = (TH1F*)TDirectory::CurrentDirectory()->Get("hDiv");
    hDiv->SetStats(0);
    TH1F *hDivFR = (TH1F*)hDiv->Clone("hDivFR");
    T->Draw("Division>>hDivFR","Nation==\"FR\"","goff");
diff -ur root-5.32.00.orig/tutorials/hist/hstack.C root-5.32.00/tutorials/hist/hstack.C
--- root-5.32.00.orig/tutorials/hist/hstack.C	2011-12-02 12:41:49.000000000 +0100
+++ root-5.32.00/tutorials/hist/hstack.C	2012-02-11 16:08:54.067671744 +0100
@@ -33,18 +33,18 @@
    // in top right pad, draw the stack in non-stack mode 
    // and errors option
    cst->cd(2);
-   gPad->SetGrid();
+   TVirtualPad::Pad()->SetGrid();
    hs->Draw("nostack,e1p");
    //in bottom left, draw in stack mode with "lego1" option
    cst->cd(3);
-   gPad->SetFrameFillColor(17);
-   gPad->SetTheta(3.77);
-   gPad->SetPhi(2.9);
+   TVirtualPad::Pad()->SetFrameFillColor(17);
+   TVirtualPad::Pad()->SetTheta(3.77);
+   TVirtualPad::Pad()->SetPhi(2.9);
    hs->Draw("lego1");
 
    cst->cd(4);
    //create two 2-D histograms and draw them in stack mode
-   gPad->SetFrameFillColor(17);
+   TVirtualPad::Pad()->SetFrameFillColor(17);
    THStack *a = new THStack("a","Stacked 2D histograms");
    TF2 *f1 = new TF2("f1",
       "xygaus + xygaus(5) + xylandau(10)",-4,4,-4,4);
diff -ur root-5.32.00.orig/tutorials/hsimple.C root-5.32.00/tutorials/hsimple.C
--- root-5.32.00.orig/tutorials/hsimple.C	2011-12-02 12:41:50.000000000 +0100
+++ root-5.32.00/tutorials/hsimple.C	2012-02-11 15:11:23.468765021 +0100
@@ -25,7 +25,7 @@
 //  write access to this directory, otherwise the file is created in $PWD
 
    TString filename = "hsimple.root";
-   TString dir = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
+   TString dir = gSystem->UnixPathName(TInterpreter::Instance()->GetCurrentMacroName());
    dir.ReplaceAll("hsimple.C","");
    dir.ReplaceAll("/./","/");
    TFile *hfile = 0;
diff -ur root-5.32.00.orig/tutorials/math/chi2test.C root-5.32.00/tutorials/math/chi2test.C
--- root-5.32.00.orig/tutorials/math/chi2test.C	2011-12-02 12:41:49.000000000 +0100
+++ root-5.32.00/tutorials/math/chi2test.C	2012-02-11 15:11:23.535764169 +0100
@@ -149,25 +149,25 @@
 
 // Draw Histogramms and Graphs
   c1->cd(1);
-  gPad->SetFrameFillColor(21);
+  TVirtualPad::Pad()->SetFrameFillColor(21);
   h1->SetMarkerColor(4);
   h1->SetMarkerStyle(20);
 
   h1->Draw("E");
 
   c1->cd(2);
-  gPad->SetFrameFillColor(21);
+  TVirtualPad::Pad()->SetFrameFillColor(21);
   h2->Draw("");
   h2->SetMarkerColor(4);
   h2->SetMarkerStyle(20);
 
   c1->cd(3);
-  gPad->SetFrameFillColor(21);
-  gPad->SetGridy();
+  TVirtualPad::Pad()->SetFrameFillColor(21);
+  TVirtualPad::Pad()->SetGridy();
   resgr->Draw("APL");
 
   c1->cd(4);
-  gPad->SetFrameFillColor(21);
+  TVirtualPad::Pad()->SetFrameFillColor(21);
   qqplot->Draw("AP");
 
   c1->cd(0);
diff -ur root-5.32.00.orig/tutorials/tree/cernbuild.C root-5.32.00/tutorials/tree/cernbuild.C
--- root-5.32.00.orig/tutorials/tree/cernbuild.C	2012-02-10 14:59:26.357917020 +0100
+++ root-5.32.00/tutorials/tree/cernbuild.C	2012-02-11 17:52:20.988688923 +0100
@@ -19,7 +19,7 @@
    //The input file cern.dat is a copy of the CERN staff data base
    //from 1988
    TString filename = "cernstaff.root";
-   TString dir = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
+   TString dir = gSystem->UnixPathName(TInterpreter::Instance()->GetCurrentMacroName());
    dir.ReplaceAll("cernbuild.C","");
    dir.ReplaceAll("/./","/");
    FILE *fp = fopen(Form("%scernstaff.dat",dir.Data()),"r");
diff -ur root-5.32.00.orig/tutorials/tree/cernstaff.C root-5.32.00/tutorials/tree/cernstaff.C
--- root-5.32.00.orig/tutorials/tree/cernstaff.C	2011-12-02 12:41:49.000000000 +0100
+++ root-5.32.00/tutorials/tree/cernstaff.C	2012-02-11 16:02:20.534771748 +0100
@@ -10,17 +10,17 @@
    c1->SetFillColor(17);
    c1->Divide(2,2);
    // make table of number of people per Nation & Division
-   c1->cd(1); gPad->SetGrid();
+   c1->cd(1); TVirtualPad::Pad()->SetGrid();
    T->Draw("Nation:Division>>hN","","text");
-   TH2F *hN = (TH2F*)gDirectory->Get("hN");
+   TH2F *hN = (TH2F*)TDirectory::CurrentDirectory()->Get("hN");
    hN->SetMarkerSize(1.6);
    hN->SetStats(0);
    
    //make profile of Average cost per Nation
-   c1->cd(2); gPad->SetGrid();
-   gPad->SetLeftMargin(0.12);
+   c1->cd(2); TVirtualPad::Pad()->SetGrid();
+   TVirtualPad::Pad()->SetLeftMargin(0.12);
    T->Draw("Cost:Nation>>hNation","","prof,goff");
-   TH1F *hNation = (TH1F*)gDirectory->Get("hNation");
+   TH1F *hNation = (TH1F*)TDirectory::CurrentDirectory()->Get("hNation");
    hNation->SetTitle("Average Cost per Nation");
    hNation->LabelsOption(">"); //sort by decreasing bin contents
    hNation->SetMaximum(13000);
@@ -30,7 +30,7 @@
    hNation->Draw();
    
    //make stacked plot of Nations versus Grade
-   c1->cd(3); gPad->SetGrid();
+   c1->cd(3); TVirtualPad::Pad()->SetGrid();
    THStack *hGrades = new THStack("hGrades","Nations versus Grade");
    TH1F *hFR = new TH1F("hFR","FR",12,3,15);
     hFR->SetFillColor(kCyan);
@@ -62,10 +62,10 @@
    legend->Draw();
    
    //make histogram of age distribution
-   c1->cd(4); gPad->SetGrid();
+   c1->cd(4); TVirtualPad::Pad()->SetGrid();
    T->Draw("Age");
    T->Draw("Age>>hRetired","Age>(65-2002+1988)","same");
-   TH1F *hRetired = (TH1F*)gDirectory->Get("hRetired");
+   TH1F *hRetired = (TH1F*)TDirectory::CurrentDirectory()->Get("hRetired");
    hRetired->SetFillColor(kRed);
    hRetired->SetFillStyle(3010);
    
