diff -ur root-5.34.17.orig/core/base/src/TTimeStamp.cxx root-5.34.17/core/base/src/TTimeStamp.cxx
--- root-5.34.17.orig/core/base/src/TTimeStamp.cxx	2014-02-24 09:37:53.000000000 +0100
+++ root-5.34.17/core/base/src/TTimeStamp.cxx	2014-02-27 09:37:28.381639337 +0100
@@ -813,7 +813,7 @@
           tmstruct.tm_isdst);
 #if (defined(linux) && !defined(R__WINGCC)) || defined(R__MACOSX)
    printf(",\n      tm_gmtoff %6ld, tm_zone \"%s\"",
-#if defined(__USE_BSD) || defined(R__MACOSX)
+#if defined(__USE_MISC) || defined(R__MACOSX)
    tmstruct.tm_gmtoff, tmstruct.tm_zone);
 #else
    tmstruct.__tm_gmtoff, tmstruct.__tm_zone);
