diff -ur root-5.26.00d.orig/tmva/inc/MsgLogger.h root-5.26.00d/tmva/inc/MsgLogger.h
--- root-5.26.00d.orig/tmva/inc/MsgLogger.h	2010-08-24 13:30:17.000000000 +0200
+++ root-5.26.00d/tmva/inc/MsgLogger.h	2010-08-31 19:44:33.000000000 +0200
@@ -100,7 +100,7 @@
 
       // private utility routines
       void Send();
-      void InitMaps();
+      static void InitMaps();
       void WriteMsg( EMsgType type, const std::string& line ) const;
 
       const TObject*                  fObjSource;      // the source TObject (used for name)
diff -ur root-5.26.00d.orig/tmva/src/MsgLogger.cxx root-5.26.00d/tmva/src/MsgLogger.cxx
--- root-5.26.00d.orig/tmva/src/MsgLogger.cxx	2010-08-24 13:30:17.000000000 +0200
+++ root-5.26.00d/tmva/src/MsgLogger.cxx	2010-08-31 19:42:16.000000000 +0200
@@ -54,7 +54,6 @@
      fMinType   ( minType )
 {
    // constructor
-   InitMaps();
 }
 
 //_______________________________________________________________________
@@ -65,7 +64,6 @@
      fMinType   ( minType )
 {
    // constructor
-   InitMaps();
 }
 
 //_______________________________________________________________________
@@ -76,7 +74,6 @@
      fMinType   ( minType )
 {
    // constructor
-   InitMaps();
 }
 
 //_______________________________________________________________________
@@ -86,7 +83,6 @@
      TObject()
 {
    // copy constructor
-   InitMaps();
    *this = parent;
 }
 
@@ -178,6 +174,7 @@
 
    if (type < fMinType || (fgInhibitOutput && type!=kFATAL)) return; // no output
 
+   InitMaps();
    std::map<EMsgType, std::string>::const_iterator stype;
    if ((stype = fgTypeMap.find( type )) == fgTypeMap.end()) return;
    if (!gConfig().IsSilent() || type==kFATAL) {
