diff -ur root-6.08.02.orig/documentation/doxygen/filter.cxx root-6.08.02/documentation/doxygen/filter.cxx
--- root-6.08.02.orig/documentation/doxygen/filter.cxx	2016-12-03 00:07:29.000000000 +0100
+++ root-6.08.02/documentation/doxygen/filter.cxx	2016-12-06 20:59:12.642635390 +0100
@@ -306,18 +306,9 @@
 
       // notebook found
       if (gLineString.find("\\notebook") != string::npos) {
-         ExecuteCommand(StringFormat("python converttonotebook.py %s %s/notebooks/",
-                                          gFileName.c_str(), gOutDir.c_str()));
-
-         if (gPython){
-             gLineString = "## ";
-         }
-         else{
-             gLineString = "/// ";
-         }
-         gLineString += StringFormat( "\\htmlonly <a href=\"http://nbviewer.jupyter.org/url/root.cern.ch/doc/master/notebooks/%s.nbconvert.ipynb\" target=\"_blank\"><img src= notebook.gif alt=\"View in nbviewer\" style=\"height:1em\" ></a> <a href=\"https://cern.ch/swanserver/cgi-bin/go?projurl=https://root.cern.ch/doc/master/notebooks/%s.nbconvert.ipynb\" target=\"_blank\"><img src=\"http://swanserver.web.cern.ch/swanserver/images/badge_swan_white_150.png\"  alt=\"Open in SWAN\" style=\"height:1em\" ></a> \\endhtmlonly \n", gMacroName.c_str() , gMacroName.c_str());
-
+         gLineString = "";
       }
+
       // \macro_output found
       if (gLineString.find("\\macro_output") != string::npos) {
          if (!gPython) ExecuteCommand(StringFormat("root -l -b -q %s", gFileName.c_str()).c_str());
