diff -ur root-6.22.00.orig/bindings/pyroot_legacy/ROOT.py root-6.22.00/bindings/pyroot_legacy/ROOT.py
--- root-6.22.00.orig/bindings/pyroot_legacy/ROOT.py	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/bindings/pyroot_legacy/ROOT.py	2020-06-29 09:07:41.574798580 +0200
@@ -438,9 +438,7 @@
 # This function is injected as method to the respective classes in Pythonize.cxx.
 _root._RDataFrameAsNumpy = _RDataFrameAsNumpy
 
-# This Pythonisation is there only for 64 bits builds
-if (sys.maxsize > 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-    _root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
+_root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
 
 
 ### RINT command emulation ------------------------------------------------------
diff -ur root-6.22.00.orig/build/unix/makepchinput.py root-6.22.00/build/unix/makepchinput.py
--- root-6.22.00.orig/build/unix/makepchinput.py	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/build/unix/makepchinput.py	2020-06-29 09:07:41.639798769 +0200
@@ -252,9 +252,6 @@
                            "math/vdt",
                            "tmva/rmva"]
 
-   if (sys.platform != 'win32' and sys.maxsize <= 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-      PCHPatternsBlacklist.append("tree/dataframe")
-
    accepted = isAnyPatternInString(PCHPatternsWhitelist,dirName) and \
                not isAnyPatternInString(PCHPatternsBlacklist,dirName)
 
diff -ur root-6.22.00.orig/tree/dataframe/test/dataframe_cache.cxx root-6.22.00/tree/dataframe/test/dataframe_cache.cxx
--- root-6.22.00.orig/tree/dataframe/test/dataframe_cache.cxx	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tree/dataframe/test/dataframe_cache.cxx	2020-06-29 09:07:41.640798772 +0200
@@ -224,8 +224,6 @@
 
 }
 
-#ifdef R__B64
-
 TEST(Cache, Regex)
 {
 
@@ -335,8 +333,6 @@
    gSystem->Unlink(fileName);
 }
 
-#endif // R__B64
-
 // ROOT-10563
 TEST(Cache, Alias)
 {
diff -ur root-6.22.00.orig/tree/dataframe/test/datasource_arrow.cxx root-6.22.00/tree/dataframe/test/datasource_arrow.cxx
--- root-6.22.00.orig/tree/dataframe/test/datasource_arrow.cxx	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tree/dataframe/test/datasource_arrow.cxx	2020-06-29 09:07:41.645798787 +0200
@@ -177,8 +177,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(RArrowDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RArrowDS(createTestTable(), {}));
@@ -250,5 +248,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.22.00.orig/tree/dataframe/test/datasource_csv.cxx root-6.22.00/tree/dataframe/test/datasource_csv.cxx
--- root-6.22.00.orig/tree/dataframe/test/datasource_csv.cxx	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tree/dataframe/test/datasource_csv.cxx	2020-06-29 09:07:41.645798787 +0200
@@ -195,8 +195,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(RCsvDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RCsvDS(fileName0));
@@ -292,5 +290,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.22.00.orig/tree/dataframe/test/datasource_root.cxx root-6.22.00/tree/dataframe/test/datasource_root.cxx
--- root-6.22.00.orig/tree/dataframe/test/datasource_root.cxx	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tree/dataframe/test/datasource_root.cxx	2020-06-29 09:07:41.649798799 +0200
@@ -117,8 +117,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(TRootTDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RRootDS(treeName, fileGlob));
@@ -190,5 +188,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.22.00.orig/tree/dataframe/test/datasource_trivial.cxx root-6.22.00/tree/dataframe/test/datasource_trivial.cxx
--- root-6.22.00.orig/tree/dataframe/test/datasource_trivial.cxx	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tree/dataframe/test/datasource_trivial.cxx	2020-06-29 09:07:41.654798814 +0200
@@ -132,8 +132,6 @@
    EXPECT_EQ(*tdfAll.Count(), 20ULL);
 }
 
-#ifdef R__B64
-
 TEST(RTrivialDS, FromARDFWithJitting)
 {
    std::unique_ptr<RDataSource> tds(new RTrivialDS(32));
@@ -236,5 +234,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.22.00.orig/tutorials/CMakeLists.txt root-6.22.00/tutorials/CMakeLists.txt
--- root-6.22.00.orig/tutorials/CMakeLists.txt	2020-06-14 17:51:48.000000000 +0200
+++ root-6.22.00/tutorials/CMakeLists.txt	2020-06-29 09:07:41.655798816 +0200
@@ -257,10 +257,6 @@
   set(root7_veto v7/ntuple/ntpl004_dimuon.C)
 endif()
 
-if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-  set(bits32_veto dataframe/*.C graphs/timeSeriesFrom*.C v7/ntuple/ntpl004_dimuon.C)
-endif()
-
 #---These ones are disabled !!! ------------------------------------
 set(extra_veto
   legacy/benchmarks.C
@@ -313,7 +309,6 @@
              ${classic_veto}
              ${pythia_veto}
              ${root7_veto}
-             ${bits32_veto}
              ${xrootd_veto}
              ${mlp_veto}
              ${spectrum_veto}
@@ -526,7 +521,6 @@
   list(REMOVE_ITEM pytutorials ${pyveto})
 
   if(NOT dataframe)
-    set(dataframe_veto_py dataframe/*.py)
     file(GLOB dataframe_veto_py RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} dataframe/*.py tmva/tmva*.py)
     list(REMOVE_ITEM pytutorials ${dataframe_veto_py})
     list(REMOVE_ITEM pytutorials
