diff -ur root-6.18.00.orig/bindings/pyroot/ROOT.py root-6.18.00/bindings/pyroot/ROOT.py
--- root-6.18.00.orig/bindings/pyroot/ROOT.py	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/bindings/pyroot/ROOT.py	2019-06-26 15:14:34.684257765 +0200
@@ -449,9 +449,7 @@
 # This function is injected as method to the respective classes in Pythonize.cxx.
 _root._RDataFrameAsNumpy = _RDataFrameAsNumpy
 
-# This Pythonisation is there only for 64 bits builds
-if (sys.maxsize > 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-    _root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
+_root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
 
 
 ### RINT command emulation ------------------------------------------------------
diff -ur root-6.18.00.orig/bindings/pyroot/test/CMakeLists.txt root-6.18.00/bindings/pyroot/test/CMakeLists.txt
--- root-6.18.00.orig/bindings/pyroot/test/CMakeLists.txt	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/bindings/pyroot/test/CMakeLists.txt	2019-06-26 15:22:03.843577496 +0200
@@ -4,10 +4,7 @@
 ROOT_ADD_PYUNITTEST(pyroot_rvec rvec.py)
 if(NUMPY_FOUND)
   ROOT_ADD_PYUNITTEST(pyroot_array_interface array_interface.py)
-  # this excludes RDF for 32 bits builds because of clang/gcc abi issues
-  if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    ROOT_ADD_PYUNITTEST(pyroot_rdataframe_asnumpy rdataframe_asnumpy.py)
-    ROOT_ADD_PYUNITTEST(pyroot_ttree_asmatrix ttree_asmatrix.py)
-  endif()
+  ROOT_ADD_PYUNITTEST(pyroot_rdataframe_asnumpy rdataframe_asnumpy.py)
+  ROOT_ADD_PYUNITTEST(pyroot_ttree_asmatrix ttree_asmatrix.py)
 endif()
 
diff -ur root-6.18.00.orig/build/unix/makepchinput.py root-6.18.00/build/unix/makepchinput.py
--- root-6.18.00.orig/build/unix/makepchinput.py	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/build/unix/makepchinput.py	2019-06-26 15:14:34.760256972 +0200
@@ -263,9 +263,6 @@
                            "math/vdt",
                            "tmva/rmva"]
 
-   if (sys.platform != 'win32' and sys.maxsize <= 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-      PCHPatternsBlacklist.append("tree/dataframe")
-
    accepted = isAnyPatternInString(PCHPatternsWhitelist,dirName) and \
                not isAnyPatternInString(PCHPatternsBlacklist,dirName)
 
diff -ur root-6.18.00.orig/cmake/scripts/ROOTConfig.cmake.in root-6.18.00/cmake/scripts/ROOTConfig.cmake.in
--- root-6.18.00.orig/cmake/scripts/ROOTConfig.cmake.in	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/cmake/scripts/ROOTConfig.cmake.in	2019-06-26 15:14:34.768256889 +0200
@@ -83,7 +83,7 @@
   set(CMAKE_FIND_LIBRARY_PREFIXES "lib")
   set(CMAKE_FIND_LIBRARY_SUFFIXES ".lib" ".dll")
 endif()
-foreach(_cpt Core Imt RIO Net Hist Graf Graf3d Gpad ROOTDataFrame Tree TreePlayer Rint Postscript Matrix Physics MathCore Thread MultiProc ${ROOT_FIND_COMPONENTS})
+foreach(_cpt Core Imt RIO Net Hist Graf Graf3d Gpad ROOTDataFrame ROOTVecOps Tree TreePlayer Rint Postscript Matrix Physics MathCore Thread MultiProc ${ROOT_FIND_COMPONENTS})
   find_library(ROOT_${_cpt}_LIBRARY ${_cpt} HINTS ${ROOT_LIBRARY_DIR})
   if(ROOT_${_cpt}_LIBRARY)
     mark_as_advanced(ROOT_${_cpt}_LIBRARY)
diff -ur root-6.18.00.orig/config/root-config.in root-6.18.00/config/root-config.in
--- root-6.18.00.orig/config/root-config.in	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/config/root-config.in	2019-06-26 15:41:39.518330743 +0200
@@ -72,15 +72,10 @@
 newlib="-lNew"
 rootglibs="-lGui"
 rootevelibs="-lEve -lEG -lGeom -lGed -lRGL"
-rootlibs="-lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTVecOps -lTree -lTreePlayer\
+rootlibs="-lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTDataFrame -lROOTVecOps -lTree -lTreePlayer\
           -lRint -lPostscript -lMatrix -lPhysics -lMathCore -lThread\
           -lMultiProc"
 
-if [ "@UNIX@" != "1" ] || [ "@CMAKE_SIZEOF_VOID_P@" != "4" ]; then
-   rootlibs="$rootlibs -lROOTDataFrame"
-fi
-
-
 if test "$platform" = "win32"; then
    rootulibs="-include:_G__cpp_setupG__Net        \
               -include:_G__cpp_setupG__IO         \
diff -ur root-6.18.00.orig/tree/dataframe/CMakeLists.txt root-6.18.00/tree/dataframe/CMakeLists.txt
--- root-6.18.00.orig/tree/dataframe/CMakeLists.txt	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/tree/dataframe/CMakeLists.txt	2019-06-26 15:14:34.827256273 +0200
@@ -3,11 +3,6 @@
 # @author Danilo Piparo CERN, Pere Mato CERN
 ############################################################################
 
-# Disable RDataFrame on 32-bit UNIX platforms due to ROOT-9236
-if(UNIX AND CMAKE_SIZEOF_VOID_P EQUAL 4)
-  return()
-endif()
-
 if(arrow)
   list(APPEND RDATAFRAME_EXTRA_HEADERS ROOT/RArrowDS.hxx)
   list(APPEND RDATAFRAME_EXTRA_INCLUDES -I${ARROW_INCLUDE_DIR})
diff -ur root-6.18.00.orig/tutorials/CMakeLists.txt root-6.18.00/tutorials/CMakeLists.txt
--- root-6.18.00.orig/tutorials/CMakeLists.txt	2019-06-25 10:52:11.000000000 +0200
+++ root-6.18.00/tutorials/CMakeLists.txt	2019-06-26 15:18:15.834951333 +0200
@@ -212,10 +212,6 @@
                  v7/ntuple/ntpl004_dimuon.C)
 endif()
 
-if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-  set(bits32_veto dataframe/*.C graphs/timeSeriesFrom*.C v7/ntuple/ntpl004_dimuon.C)
-endif()
-
 #---These ones are disabled !!! ------------------------------------
 set(extra_veto
   htmlex.C
@@ -266,7 +262,6 @@
              ${classic_veto}
              ${pythia_veto}
              ${root7_veto}
-             ${bits32_veto}
              ${xrootd_veto}
              )
 
@@ -424,13 +419,6 @@
 
   list(REMOVE_ITEM pytutorials ${pyveto})
 
-  if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-    set(bits32_veto_py dataframe/*.py)
-    file(GLOB bits32_veto_py RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} dataframe/*.py)
-    list(REMOVE_ITEM pytutorials ${bits32_veto_py})
-    list(REMOVE_ITEM pytutorials pyroot/pyroot002_TTreeAsMatrix.py)
-  endif()
-
   #---Python tutorials dependencies--------------------------------------
   set(pyroot-ntuple1-depends tutorial-pyroot-hsimple-py)
   set(pyroot-h1draw-depends tutorial-pyroot-hsimple-py)
