From 6991169440211dd29c4608541b8740337c1891fa Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Wed, 27 Sep 2017 21:06:41 +0200
Subject: [PATCH] Fix for new mysql_config

---
 cmake/modules/FindMySQL.cmake | 9 +++------
 configure                     | 4 +++-
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/cmake/modules/FindMySQL.cmake b/cmake/modules/FindMySQL.cmake
index 123505a393..c818f5a1ab 100644
--- a/cmake/modules/FindMySQL.cmake
+++ b/cmake/modules/FindMySQL.cmake
@@ -20,12 +20,9 @@ endif()
 if(MYSQL_CONFIG_EXECUTABLE)
   execute_process(COMMAND ${MYSQL_CONFIG_EXECUTABLE} --cflags OUTPUT_VARIABLE MYSQL_CFLAGS OUTPUT_STRIP_TRAILING_WHITESPACE)
   separate_arguments(MYSQL_CFLAGS)
-  string( REGEX MATCH "-I[^;]+" MYSQL_INCLUDE_DIR "${MYSQL_CFLAGS}" )
-  string( REPLACE "-I" "" MYSQL_INCLUDE_DIR "${MYSQL_INCLUDE_DIR}")
-  if(NOT EXISTS ${MYSQL_INCLUDE_DIR})
-    set(MYSQL_INCLUDE_DIR MYSQL_INCLUDE_DIR-NOTFOUND)
-  endif()
-  string( REGEX REPLACE "-I[^;]+;" "" MYSQL_CFLAGS "${MYSQL_CFLAGS}" )
+  string( REGEX MATCHALL "-I[^;]+" MYSQL_INCLUDE_DIRS "${MYSQL_CFLAGS}" )
+  string( REPLACE "-I" "" MYSQL_INCLUDE_DIRS "${MYSQL_INCLUDE_DIRS}")
+  find_path(MYSQL_INCLUDE_DIR mysql.h PATHS ${MYSQL_INCLUDE_DIRS} NO_DEFAULT_PATH)
   execute_process(COMMAND ${MYSQL_CONFIG_EXECUTABLE} --libs OUTPUT_VARIABLE MYSQL_LIBRARIES OUTPUT_STRIP_TRAILING_WHITESPACE)
 else()
   find_path(MYSQL_INCLUDE_DIR mysql.h
diff --git a/configure b/configure
index 26fe09ab31..6bdf5b29c4 100755
--- a/configure
+++ b/configure
@@ -3738,9 +3738,11 @@ if test ! "x$enable_mysql" = "xno"; then
           else
              result "ok"
              mysqllib=`$mysql_config --libs | sed -e s/\'//g`
-             mysqlincdir=`$mysql_config --cflags | sed -e 's,^.*-I\([^ ]*\).*$,\1,' -e s/\'//g`
              mysqllibdir=""
+             mysqlincdir=`$mysql_config --cflags | sed -e s/\'//g | tr ' ' '\n' | grep ^-I | sed -e 's,^-I,,' | tr '\n' ' '`
              check_header "mysql.h" "" $mysqlincdir
+             mysqlinc=$found_hdr
+             mysqlincdir=$found_dir
              if test "x$found_hdr" = "x"; then
                 enable_mysql="no"
              fi
-- 
2.13.5

