diff -ur root-6.06.02.orig/test/CMakeLists.txt root-6.06.02/test/CMakeLists.txt
--- root-6.06.02.orig/test/CMakeLists.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/test/CMakeLists.txt	2016-04-02 17:06:39.536489046 +0200
@@ -239,13 +239,14 @@
 
 #--stressProof-------------------------------------------------------------------------------
 if(NOT WIN32)
-  add_custom_target(TestData ALL COMMAND ${CMAKE_COMMAND} -DDST=${CMAKE_SOURCE_DIR}/files -P ${CMAKE_CURRENT_SOURCE_DIR}/rootDownloadData.cmake)
+  add_custom_target(TestData COMMAND ${CMAKE_COMMAND} -DDST=${CMAKE_SOURCE_DIR}/files -P ${CMAKE_CURRENT_SOURCE_DIR}/rootDownloadData.cmake)
   ROOT_EXECUTABLE(stressProof stressProof.cxx LIBRARIES Proof ProofPlayer Hist)
   ROOT_ADD_TEST(test-stressproof COMMAND stressProof lite://
                                          -h1 ${CMAKE_SOURCE_DIR}/files/h1 -event ${CMAKE_SOURCE_DIR}/files/event
                                          -l /tmp/stressProof-%d.log -cleanlog -catlog -noprogress
                                          ENVIRONMENT PATH=${CMAKE_BINARY_DIR}/bin:$ENV{PATH}
-                                         FAILREGEX "FAILED|Error in")
+                                         FAILREGEX "FAILED|Error in"
+                                         DEPENDS TestData)
 endif()
 
 #--testbits----------------------------------------------------------------------------------
