diff --git a/math/mathmore/test/testSpecFunc.cxx b/math/mathmore/test/testSpecFunc.cxx
index d768160..8c223ee 100644
--- a/math/mathmore/test/testSpecFunc.cxx
+++ b/math/mathmore/test/testSpecFunc.cxx
@@ -186,7 +186,7 @@ int testSpecFunc() {
 
    iret |= compare("hyperg(8, -8, 1, 0.5) ", hyperg(8, -8, 1, 0.5), 0.13671875);
 
-   iret |= compare("laguerre(4, 1.) ", laguerre(4, 1.), -0.6250); // need to find more precise value
+   iret |= compare("laguerre(4, 1.) ", laguerre(4, 1.), -0.6250, 4); // need to find more precise value
 
    iret |= compare("legendre(10, -0.5) ", legendre(10, -0.5), -0.1882286071777345);
 
diff --git a/test/stressMathCore.cxx b/test/stressMathCore.cxx
index 1006dd1..b862f36 100644
--- a/test/stressMathCore.cxx
+++ b/test/stressMathCore.cxx
@@ -1117,7 +1117,7 @@ int testVector(int ngen, bool testio=false) {
 
    s1 = a.testOperations(v1);  a.print(VecType<V1>::name()+" operations");
    scale = Dim*20;
-   if (Dim==3 && VecType<V2>::name() == "RhoEtaPhiVector") scale *= 10; // for problem with RhoEtaPhi
+   if (Dim==3 && VecType<V2>::name() == "RhoEtaPhiVector") scale *= 12; // for problem with RhoEtaPhi
    if (Dim==4 && VecType<V2>::name() == "PtEtaPhiMVector") scale *= 10;
 #if defined (R__LINUX) && !defined(R__B64)
    // problem of precision on linux 32
