diff -ur root-6.06.02.orig/misc/table/inc/TCernLib.h root-6.06.02/misc/table/inc/TCernLib.h
--- root-6.06.02.orig/misc/table/inc/TCernLib.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/misc/table/inc/TCernLib.h	2016-03-07 19:15:34.726025891 +0100
@@ -15,7 +15,7 @@
 #include "Rtypes.h"
 #include <string.h>
 
-// http://wwwinfo.cern.ch/asdoc/shortwrupsdir/f110/top.html
+// https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html
 
 ///////////////////////////////////////////////////////////////////////////////////////
 //                                                                                   //
@@ -25,8 +25,9 @@
 // its negative, or a multiple of it, transpose a given matrix, build up a unit      //
 // matrix, multiply a matrix by a diagonal (from left or from right) and may         //
 // add the result to another matrix, add to square matrix the multiple of a diagonal //
-// matrix, compute the products <IMG WIDTH=79 HEIGHT=12 ALIGN=BOTTOM ALT="tex2html_wrap_inline191" SRC="gif/mxpack_ABAt.gif"> (<IMG WIDTH=16 HEIGHT=12 ALIGN=BOTTOM ALT="tex2html_wrap_inline193" SRC="gif/mxpack_At.gif"> denotes the transpose of <IMG WIDTH=1
-// It is assumed that matrices are begin_html <B>row-wise without gaps</B> end_html without gaps.                     //
+// matrix, compute the products X=ABA' (A' denotes the transpose of A) and X=A'BA.   //
+// It is assumed that matrices are <B>row-wise without gaps</B>, contrary to the     //
+// Fortran convention.                                                               //
 //                                                                                   //
 ///////////////////////////////////////////////////////////////////////////////////////
 
diff -ur root-6.06.02.orig/misc/table/src/TCernLib.cxx root-6.06.02/misc/table/src/TCernLib.cxx
--- root-6.06.02.orig/misc/table/src/TCernLib.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/misc/table/src/TCernLib.cxx	2016-03-07 19:15:34.728025914 +0100
@@ -11,8 +11,8 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 // The set of methods to work with the plain matrix / vector
-// "derived" from  http://wwwinfo.cern.ch/asdoc/shortwrupsdir/f110/top.html
-// "derived" from  http://wwwinfo.cern.ch/asdoc/shortwrupsdir/f112/top.html
+// "derived" from  https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html
+// "derived" from  https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html
 //
 // Revision 1.7  2006/05/21 18:05:26  brun
 // Fix more coding conventions violations
@@ -33,7 +33,7 @@
 //
 // Revision 1.3  2003/04/03 17:39:39  fine
 // Make merge with ROOT 3.05.03 and add TR package
-//122
+//
 // Revision 1.2  2003/02/04 23:35:20  fine
 // Clean up
 //
@@ -154,27 +154,24 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Matrix Multiplication
+///
 /// CERN PROGLIB# F110    MXMLRT          .VERSION KERNFOR  2.00  720707
 /// ORIG. 01/01/64 RKB
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F110](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html)
 
 float *TCL::mxmlrt_0_(int n__, const float *a, const float *b, float *c, int ni,int nj)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f110/top.html">F110</A>
- <!--*/
- // -->END_HTML
-
+// --      ENTRY MXMLRT
+// --                C = A(I,J) X B(J,J) X A*(J,I)
+// --                A* TANDS FOR A-TRANSPOSED
+//             mxmlrt (A,B,C,NI,NJ)     IS EQUIVALENT TO
+//             CALL MXMPY (A,B,X,NI,NJ,NJ)
+//             CALL MXMPY1 (X,A,C,NI,NJ,NI)
 
-// --      ENTRY MXMLRT */
-// --                C = A(I,J) X B(J,J) X A*(J,I) */
-// --                A* TANDS FOR A-TRANSPOSED */
-//             mxmlrt (A,B,C,NI,NJ)     IS EQUIVALENT TO */
-//             CALL MXMPY (A,B,X,NI,NJ,NJ) */
-//             CALL MXMPY1 (X,A,C,NI,NJ,NI) */
-
-/*        OR   CALL MXMPY1 (B,A,Y,NJ,NJ,NI) */
-/*             CALL MXMPY (A,Y,C,NI,NJ,NI) */
+//        OR   CALL MXMPY1 (B,A,Y,NJ,NJ,NI)
+//             CALL MXMPY (A,Y,C,NI,NJ,NI)
 
 
 // --                C = A*(I,J) X B(J,J) X A(J,I)
@@ -191,6 +188,12 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Matrix Multiplication (double precision)
+///
+/// CERN PROGLIB# F110    MXMLRT          .VERSION KERNFOR  2.00  720707
+/// ORIG. 01/01/64 RKB
+///
+/// See original documentation of CERNLIB package
+/// [F110](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html)
 
 double *TCL::mxmlrt_0_(int n__, const double *a, const double *b, double *c, int ni,int nj)
 {
@@ -215,36 +218,31 @@
     for (int l = 1; l <= i; ++l,ia += j,++ib) b[ib] = a[ia]; }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Matrix Transposition
 ///
-///  Matrix Transposition
 /// CERN PROGLIB# F110    MXTRP           .VERSION KERNFOR  1.0   650809
 /// ORIG. 01/01/64 RKB
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F110](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html)
 
 float *TCL::mxtrp(const float *a, float *b, int i, int j)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f110/top.html">F110</A>
- <!--*/
- // -->END_HTML
-
    TCL_MXTRP(a, b, i, j)
    return b;
 } /* mxtrp */
 
 ////////////////////////////////////////////////////////////////////////////////
-///  Matrix Transposition (double precision)
+/// Matrix Transposition (double precision)
+///
 /// CERN PROGLIB# F110    MXTRP           .VERSION KERNFOR  1.0   650809
 /// ORIG. 01/01/64 RKB
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F110](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f110/top.html)
 
 double *TCL::mxtrp(const double *a, double *b, int i, int j)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f110/top.html">F110</A>
- <!--*/
- // -->END_HTML
-
    TCL_MXTRP(a, b, i, j)
    return b;
 
@@ -283,38 +281,35 @@
 
 
 ////////////////////////////////////////////////////////////////////////////////
-///
 /// Symmetric Multiplication of Rectangular Matrices
+///
 /// CERN PROGLIB# F112    TRAAT           .VERSION KERNFOR  4.15  861204
-/// ORIG. 18/12/74 WH */
+/// ORIG. 18/12/74 WH
+///
 /// traat.F -- translated by f2c (version 19970219).
 ///
-///BEGIN_HTML <!--
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::traat(const float *a, float *s, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAAT(a, s, m, n)
    return s;
 } /* traat_ */
 
 ////////////////////////////////////////////////////////////////////////////////
-///  Symmetric Multiplication of Rectangular Matrices
+/// Symmetric Multiplication of Rectangular Matrices
+///
 /// CERN PROGLIB# F112    TRAAT           .VERSION KERNFOR  4.15  861204
-/// ORIG. 18/12/74 WH */
+/// ORIG. 18/12/74 WH
+///
 /// traat.F -- translated by f2c (version 19970219).
 ///
-///BEGIN_HTML <!--
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::traat(const double *a, double *s, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAAT(a, s, m, n)
    return s;
 } /* traat_ */
@@ -346,34 +341,34 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Triangular - Rectangular Multiplication
+///
 /// CERN PROGLIB# F112    TRAL            .VERSION KERNFOR  4.15  861204
 /// ORIG. 18/12/74 WH
+///
 /// tral.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::tral(const float *a, const float *u, float *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAL(a, u, b, m,  n)
    return b;
 } /* tral_ */
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Triangular - Rectangular Multiplication
+///
+/// CERN PROGLIB# F112    TRAL            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// tral.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRAL            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::tral(const double *a, const double *u, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAL(a, u, b, m,  n)
    return b;
 } /* tral_ */
@@ -405,34 +400,34 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Triangular - Rectangular Multiplication
+///
 /// CERN PROGLIB# F112    TRALT           .VERSION KERNFOR  4.15  861204
 /// ORIG. 18/12/74 WH
+///
 /// tralt.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::tralt(const float *a, const float *u, float *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRALT(a, u, b, m, n)
    return b;
 } /* tralt_ */
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Triangular - Rectangular Multiplication
+///
 /// CERN PROGLIB# F112    TRALT           .VERSION KERNFOR  4.15  861204
 /// ORIG. 18/12/74 WH
+///
 /// tralt.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::tralt(const double *a, const double *u, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRALT(a, u, b, m, n)
    return b;
 } /* tralt_ */
@@ -470,34 +465,34 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Symmetric - Rectangular Multiplication
-/// CERN PROGLIB# F112    TRAS            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
+///
+/// CERN PROGLIB# F112    TRAS            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// tras.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::tras(const float *a, const float *s, float *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAS(a, s, b, m, n)
    return b;
 } /* tras_ */
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Symmetric - Rectangular Multiplication
-/// CERN PROGLIB# F112    TRAS            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
+///
+/// CERN PROGLIB# F112    TRAS            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// tras.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::tras(const double *a, const double *s, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRAS(a, s, b, m, n)
    return b;
 } /* tras_ */
@@ -543,51 +538,51 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Transformation of Symmetric Matrix
-/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
+///
+/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// trasat.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trasat(const float *a, const float *s, float *r__, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRASAT(a, s, r__, m, n)
    return r__;
 } /* trasat_ */
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Transformation of Symmetric Matrix
-/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
+///
+/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// trasat.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trasat(const double *a, const double *s, double *r__, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRASAT(a, s, r__, m, n)
    return r__;
 } /* trasat_ */
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Transformation of Symmetric Matrix
-/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
+///
+/// CERN PROGLIB# F112    TRASAT          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
 /// trasat.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trasat(const double *a, const float *s, float *r__, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    TCL_TRASAT(a, s, r__, m, n)
    return r__;
 } /* trasat_ */
@@ -596,17 +591,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trata.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRATA           .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRATA           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trata(const float *a, float *r__, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Local variables */
    int i__, j, ia, mn, ir, iat;
    double sum;
@@ -636,22 +629,21 @@
    return r__;
 } /* trata_ */
 
-//____________________________________________________________
-// trats.F -- translated by f2c (version 19970219).
+////////////////////////////////////////////////////////////////////////////////
+/// trats.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRATS           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
 float *TCL::trats(const float *a, const float *s, float *b, int m, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
 
-   /* CERN PROGLIB# F112    TRATS           .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
    /* Parameter adjuTments */
    --b;    --s;    --a;
 
@@ -684,25 +676,22 @@
    return b;
 } /* trats_ */
 
-//____________________________________________________________
-// tratsa.F -- translated by f2c (version 19970219).
-/* Subroutine */float *TCL::tratsa(const float *a, const float *s, float *r__, int m, int n)
-{
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+////////////////////////////////////////////////////////////////////////////////
+/// tratsa.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRATSA          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+float *TCL::tratsa(const float *a, const float *s, float *r__, int m, int n)
+{
    /* Local variables */
    int imax, i__, j, k;
    int ia, ir, is, iaa, ind;
    double sum;
 
-   /* CERN PROGLIB# F112    TRATSA          .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --r__;    --s;    --a;
 
@@ -743,26 +732,23 @@
    return r__;
 } /* tratsa_ */
 
-//____________________________________________________________
-// trchlu.F -- translated by f2c (version 19970219).
+////////////////////////////////////////////////////////////////////////////////
+/// trchlu.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRCHLU          .VERSION KERNFOR  4.16  870601
+/// ORIG. 18/12/74 W.HART
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
 float *TCL::trchlu(const float *a, float *b, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int ipiv, kpiv, i__, j;
    double r__, dc;
    int id, kd;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRCHLU          .VERSION KERNFOR  4.16  870601 */
-   /* ORIG. 18/12/74 W.HART */
-
-
    /* Parameter adjuTments */
    --b;    --a;
 
@@ -806,15 +792,17 @@
    return b;
 } /* trchlu_ */
 
-//____________________________________________________________
-// trchul.F -- translated by f2c (version 19970219).
-/* Subroutine */float *TCL::trchul(const float *a, float *b, int n)
+////////////////////////////////////////////////////////////////////////////////
+/// trchul.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRCHUL          .VERSION KERNFOR  4.16  870601
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
+float *TCL::trchul(const float *a, float *b, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int ipiv, kpiv, i__;
    double r__;
@@ -823,11 +811,6 @@
    int id, kd;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRCHUL          .VERSION KERNFOR  4.16  870601 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --b;    --a;
 
@@ -875,17 +858,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trinv.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRINV           .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */float *TCL::trinv(const float *t, float *s, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRINV           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+float *TCL::trinv(const float *t, float *s, int n)
+{
    int lhor, ipiv, lver, j;
    double sum = 0;
    double r__ = 0;
@@ -934,22 +915,20 @@
    return s;
 } /* trinv_ */
 
-//____________________________________________________________
-// trla.F -- translated by f2c (version 19970219).
-/* Subroutine */float *TCL::trla(const float *u, const float *a, float *b, int m, int n)
+////////////////////////////////////////////////////////////////////////////////
+/// trla.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRLA            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
+float *TCL::trla(const float *u, const float *a, float *b, int m, int n)
 {
    int ipiv, ia, ib, iu;
    double sum;
 
-   /* CERN PROGLIB# F112    TRLA            .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
-
    /* Parameter adjuTments */
    --b;    --a;    --u;
 
@@ -981,22 +960,19 @@
 } /* trla_ */
 
 ////////////////////////////////////////////////////////////////////////////////
+/// trlta.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRLTA           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* trlta.F -- translated by f2c (version 19970219).
-// Subroutine */float *TCL::trlta(const float *u, const float *a, float *b, int m, int n)
+float *TCL::trlta(const float *u, const float *a, float *b, int m, int n)
 {
    int ipiv, mxpn, i__, nTep, ia, ib, iu, mx;
    double sum;
 
-   /* CERN PROGLIB# F112    TRLTA           .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
-
    /* Parameter adjuTments */
    --b;    --a;    --u;
 
@@ -1036,16 +1012,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trpck.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRPCK           .VERSION KERNFOR  2.08  741218 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRPCK           .VERSION KERNFOR  2.08  741218
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trpck(const float *s, float *u, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    int i__, ia, ind, ipiv;
 
    /* Parameter adjuTments */
@@ -1072,17 +1047,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trqsq.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRQSQ           .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRQSQ           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trqsq(const float *q, const float *s, float *r__, int m)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    int indq, inds, imax, i__, j, k, l;
    int iq, ir, is, iqq;
    double sum;
@@ -1140,16 +1113,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsa.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSA            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRSA            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trsa(const float *s, const float *a, float *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
@@ -1190,17 +1162,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsinv.F -- translated by f2c (version 19970219).
+///
 /// CERN PROGLIB# F112    TRSINV          .VERSION KERNFOR  2.08  741218
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* Subroutine */float *TCL::trsinv(const float *g, float *gi, int n)
+float *TCL::trsinv(const float *g, float *gi, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Function Body */
    trchlu(g, gi, n);
    trinv(gi, gi, n);
@@ -1209,17 +1179,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsmlu.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSMLU          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */float *TCL::trsmlu(const float *u, float *s, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRSMLU          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+float *TCL::trsmlu(const float *u, float *s, int n)
+{
    /* Local variables */
    int lhor, lver, i__, k, l, ind;
    double sum;
@@ -1246,17 +1214,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsmul.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSMUL          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */float *TCL::trsmul(const float *g, float *gi, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRSMUL          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+float *TCL::trsmul(const float *g, float *gi, int n)
+{
    /* Local variables */
    int lhor, lver, lpiv, i__, j, k, ind;
    double sum;
@@ -1284,18 +1250,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trupck.F -- translated by f2c (version 19970219).
+///
 /// CERN PROGLIB# F112    TRUPCK          .VERSION KERNFOR  2.08  741218
 /// ORIG. 18/12/74 WH
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 float *TCL::trupck(const float *u, float *s, int m)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
-
    int i__, im, is, iu, iv, ih, m2;
 
    /* Parameter adjuTments */
@@ -1336,25 +1299,20 @@
 } /* trupck_ */
 
 ////////////////////////////////////////////////////////////////////////////////
+/// trsat.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRSAT           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* trsat.F -- translated by f2c (version 19970219).
-// Subroutine */ float *TCL::trsat(const float *s, const float *a, float *b, int m, int n)
+float *TCL::trsat(const float *s, const float *a, float *b, int m, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRSAT           .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --b;    --a;    --s;
 
@@ -1391,25 +1349,21 @@
 
 // ------  double
 
-//____________________________________________________________
-// trata.F -- translated by f2c (version 19970219).
+////////////////////////////////////////////////////////////////////////////////
+/// trata.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRATA           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
 double *TCL::trata(const double *a, double *r__, int m, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Local variables */
    int i__, j, ia, mn, ir, iat;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRATA           .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --r__;    --a;
 
@@ -1437,23 +1391,21 @@
    return 0;
 } /* trata_ */
 
-//____________________________________________________________
-// trats.F -- translated by f2c (version 19970219).
+////////////////////////////////////////////////////////////////////////////////
+/// trats.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRATS           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
 double *TCL::trats(const double *a, const double *s, double *b, int m, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRATS           .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
    /* Parameter adjuTments */
    --b;    --s;    --a;
 
@@ -1487,24 +1439,22 @@
    return 0;
 } /* trats_ */
 
-//____________________________________________________________
-// tratsa.F -- translated by f2c (version 19970219).
-/* Subroutine */double *TCL::tratsa(const double *a, const double *s, double *r__, int m, int n)
+////////////////////////////////////////////////////////////////////////////////
+/// tratsa.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRATSA          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
+double *TCL::tratsa(const double *a, const double *s, double *r__, int m, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int imax, i__, j, k;
    int ia, ir, is, iaa, ind;
    double sum;
 
-   /* CERN PROGLIB# F112    TRATSA          .VERSION KERNFOR  4.15  861204 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --r__;    --s;    --a;
 
@@ -1547,25 +1497,21 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trchlu.F -- translated by f2c (version 19970219).
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRCHLU          .VERSION KERNFOR  4.16  870601
+/// ORIG. 18/12/74 W.HART
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trchlu(const double *a, double *b, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int ipiv, kpiv, i__, j;
    double r__, dc;
    int id, kd;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRCHLU          .VERSION KERNFOR  4.16  870601 */
-   /* ORIG. 18/12/74 W.HART */
-
-
    /* Parameter adjuTments */
    --b;    --a;
 
@@ -1609,15 +1555,17 @@
    return 0;
 } /* trchlu_ */
 
-//____________________________________________________________
-// trchul.F -- translated by f2c (version 19970219).
+////////////////////////////////////////////////////////////////////////////////
+/// trchul.F -- translated by f2c (version 19970219).
+///
+/// CERN PROGLIB# F112    TRCHUL          .VERSION KERNFOR  4.16  870601
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
 double *TCL::trchul(const double *a, double *b, int n)
 {
- //BEGIN_HTML <!--
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int ipiv, kpiv, i__;
    double r__;
@@ -1626,11 +1574,6 @@
    int id, kd;
    double sum;
 
-
-   /* CERN PROGLIB# F112    TRCHUL          .VERSION KERNFOR  4.16  870601 */
-   /* ORIG. 18/12/74 WH */
-
-
    /* Parameter adjuTments */
    --b;    --a;
 
@@ -1677,17 +1620,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trinv.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRINV           .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
 ///
-///BEGIN_HTML <!--
+/// CERN PROGLIB# F112    TRINV           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trinv(const double *t, double *s, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    int lhor, ipiv, lver,  j;
    double r__;
    int mx, ndTep, ind;
@@ -1735,19 +1676,16 @@
 } /* trinv_ */
 
 ////////////////////////////////////////////////////////////////////////////////
-///
 /// trla.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRLA            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
 ///
-///BEGIN_HTML <!--
+/// CERN PROGLIB# F112    TRLA            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* Subroutine */double *TCL::trla(const double *u, const double *a, double *b, int m, int n)
+double *TCL::trla(const double *u, const double *a, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    int ipiv, ia, ib, iu;
    double sum;
 
@@ -1782,17 +1720,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trlta.F -- translated by f2c (version 19970219).
+///
 /// CERN PROGLIB# F112    TRLTA           .VERSION KERNFOR  4.15  861204
 /// ORIG. 18/12/74 WH
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trlta(const double *u, const double *a, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    int ipiv, mxpn, i__, nTep, ia, ib, iu, mx;
    double sum;
 
@@ -1834,16 +1770,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trpck.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRPCK           .VERSION KERNFOR  2.08  741218 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */double *TCL::trpck(const double *s, double *u, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRPCK           .VERSION KERNFOR  2.08  741218
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
+
+double *TCL::trpck(const double *s, double *u, int n)
+{
    int i__, ia, ind, ipiv;
 
    /* Parameter adjuTments */
@@ -1869,17 +1804,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trqsq.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRQSQ           .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRQSQ           .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trqsq(const double *q, const double *s, double *r__, int m)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    int indq, inds, imax, i__, j, k, l;
    int iq, ir, is, iqq;
    double sum;
@@ -1936,16 +1869,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsa.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSA            .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+///
+/// CERN PROGLIB# F112    TRSA            .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trsa(const double *s, const double *a, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
@@ -1985,17 +1917,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsinv.F -- translated by f2c (version 19970219).
+///
 /// CERN PROGLIB# F112    TRSINV          .VERSION KERNFOR  2.08  741218
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* Subroutine */double *TCL::trsinv(const double *g, double *gi, int n)
+double *TCL::trsinv(const double *g, double *gi, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Function Body */
    trchlu(g, gi, n);
    trinv(gi, gi, n);
@@ -2006,17 +1936,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsmlu.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSMLU          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */double *TCL::trsmlu(const double *u, double *s, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRSMLU          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+double *TCL::trsmlu(const double *u, double *s, int n)
+{
    /* Local variables */
    int lhor, lver, i__, k, l, ind;
    double sum;
@@ -2043,17 +1971,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsmul.F -- translated by f2c (version 19970219).
-/// CERN PROGLIB# F112    TRSMUL          .VERSION KERNFOR  4.15  861204 */
-/// ORIG. 18/12/74 WH */
-///BEGIN_HTML <!--
-
-/* Subroutine */double *TCL::trsmul(const double *g, double *gi, int n)
-{
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
+///
+/// CERN PROGLIB# F112    TRSMUL          .VERSION KERNFOR  4.15  861204
+/// ORIG. 18/12/74 WH
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
+double *TCL::trsmul(const double *g, double *gi, int n)
+{
    /* Local variables */
    int lhor, lver, lpiv, i__, j, k, ind;
    double sum;
@@ -2081,18 +2007,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trupck.F -- translated by f2c (version 19970219).
+///
 /// CERN PROGLIB# F112    TRUPCK          .VERSION KERNFOR  2.08  741218
 /// ORIG. 18/12/74 WH
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
-/* Subroutine */double *TCL::trupck(const double *u, double *s, int m)
+double *TCL::trupck(const double *u, double *s, int m)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
-
    int i__, im, is, iu, iv, ih, m2;
 
    /* Parameter adjuTments */
@@ -2133,17 +2056,15 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 /// trsat.F -- translated by f2c (version 19970219)
+///
 /// CERN PROGLIB# F112    TRSAT           .VERSION KERNFOR  4.15  861204
 /// ORIG. 18/12/74 WH
-///BEGIN_HTML <!--
+///
+/// See original documentation of CERNLIB package
+/// [F112](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f112/top.html)
 
 double *TCL::trsat(const double *s, const double *a, double *b, int m, int n)
 {
- /* -->
-  <b>see original documentation of CERNLIB package</b> <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f112/top.html">F112</A>
- <!--*/
- // -->END_HTML
-
    /* Local variables */
    int inds, i__, j, k, ia, ib, is;
    double sum;
