diff -ur root-6.16.00.orig/bindings/pyroot/ROOT.py root-6.16.00/bindings/pyroot/ROOT.py
--- root-6.16.00.orig/bindings/pyroot/ROOT.py	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/bindings/pyroot/ROOT.py	2019-02-12 03:57:19.824223542 +0100
@@ -379,9 +379,7 @@
     else:
         return reshaped_matrix_np
 
-# This Pythonisation is there only for 64 bits builds
-if (sys.maxsize > 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-    _root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
+_root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
 
 
 ### RINT command emulation ------------------------------------------------------
diff -ur root-6.16.00.orig/bindings/pyroot/test/CMakeLists.txt root-6.16.00/bindings/pyroot/test/CMakeLists.txt
--- root-6.16.00.orig/bindings/pyroot/test/CMakeLists.txt	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/bindings/pyroot/test/CMakeLists.txt	2019-02-11 19:30:17.443068631 +0100
@@ -4,9 +4,6 @@
 ROOT_ADD_PYUNITTEST(pyroot_rvec rvec.py)
 if(NUMPY_FOUND)
   ROOT_ADD_PYUNITTEST(pyroot_array_interface array_interface.py)
-  # this excludes RDF for 32 bits builds because of clang/gcc abi issues
-  if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    ROOT_ADD_PYUNITTEST(pyroot_ttree_asmatrix ttree_asmatrix.py)
-  endif()
+  ROOT_ADD_PYUNITTEST(pyroot_ttree_asmatrix ttree_asmatrix.py)
 endif()
 
diff -ur root-6.16.00.orig/build/unix/makepchinput.py root-6.16.00/build/unix/makepchinput.py
--- root-6.16.00.orig/build/unix/makepchinput.py	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/build/unix/makepchinput.py	2019-02-12 04:01:47.071320473 +0100
@@ -261,9 +261,6 @@
                            "math/vdt",
                            "tmva/rmva"]
 
-   if (sys.platform != 'win32' and sys.maxsize <= 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-      PCHPatternsBlacklist.append("tree/dataframe")
-
    accepted = isAnyPatternInString(PCHPatternsWhitelist,dirName) and \
                not isAnyPatternInString(PCHPatternsBlacklist,dirName)
 
diff -ur root-6.16.00.orig/cmake/scripts/ROOTConfig.cmake.in root-6.16.00/cmake/scripts/ROOTConfig.cmake.in
--- root-6.16.00.orig/cmake/scripts/ROOTConfig.cmake.in	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/cmake/scripts/ROOTConfig.cmake.in	2019-02-11 19:42:40.603045735 +0100
@@ -83,7 +83,7 @@
   set(CMAKE_FIND_LIBRARY_PREFIXES "lib")
   set(CMAKE_FIND_LIBRARY_SUFFIXES ".lib" ".dll")
 endif()
-foreach(_cpt Core Imt RIO Net Hist Graf Graf3d Gpad ROOTDataFrame Tree TreePlayer Rint Postscript Matrix Physics MathCore Thread MultiProc ${ROOT_FIND_COMPONENTS})
+foreach(_cpt Core Imt RIO Net Hist Graf Graf3d Gpad ROOTDataFrame ROOTVecOps Tree TreePlayer Rint Postscript Matrix Physics MathCore Thread MultiProc ${ROOT_FIND_COMPONENTS})
   find_library(ROOT_${_cpt}_LIBRARY ${_cpt} HINTS ${ROOT_LIBRARY_DIR})
   if(ROOT_${_cpt}_LIBRARY)
     mark_as_advanced(ROOT_${_cpt}_LIBRARY)
diff -ur root-6.16.00.orig/config/root-config.in root-6.16.00/config/root-config.in
--- root-6.16.00.orig/config/root-config.in	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/config/root-config.in	2019-02-11 19:42:38.951063533 +0100
@@ -72,15 +72,10 @@
 newlib="-lNew"
 rootglibs="-lGui"
 rootevelibs="-lEve -lEG -lGeom -lGed -lRGL"
-rootlibs="-lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTVecOps -lTree -lTreePlayer\
+rootlibs="-lCore -lImt -lRIO -lNet -lHist -lGraf -lGraf3d -lGpad -lROOTDataFrame -lROOTVecOps-lTree -lTreePlayer\
           -lRint -lPostscript -lMatrix -lPhysics -lMathCore -lThread\
           -lMultiProc"
 
-if [ `uname -m` != 'i686' ]; then
-   rootlibs="$rootlibs -lROOTDataFrame"
-fi
-
-
 if test "$platform" = "win32"; then
    rootulibs="-include:_G__cpp_setupG__Net        \
               -include:_G__cpp_setupG__IO         \
diff -ur root-6.16.00.orig/tree/dataframe/CMakeLists.txt root-6.16.00/tree/dataframe/CMakeLists.txt
--- root-6.16.00.orig/tree/dataframe/CMakeLists.txt	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/tree/dataframe/CMakeLists.txt	2019-02-11 19:26:43.236382519 +0100
@@ -3,11 +3,6 @@
 # @author Danilo Piparo CERN, Pere Mato CERN
 ############################################################################
 
-# Disable RDataFrame on 32-bit UNIX platforms due to ROOT-9236
-if(UNIX AND CMAKE_SIZEOF_VOID_P EQUAL 4)
-  return()
-endif()
-
 if(arrow)
   list(APPEND RDATAFRAME_EXTRA_HEADERS ROOT/RArrowDS.hxx)
   list(APPEND RDATAFRAME_EXTRA_INCLUDES -I${ARROW_INCLUDE_DIR})
diff -ur root-6.16.00.orig/tutorials/CMakeLists.txt root-6.16.00/tutorials/CMakeLists.txt
--- root-6.16.00.orig/tutorials/CMakeLists.txt	2019-01-23 09:23:37.000000000 +0100
+++ root-6.16.00/tutorials/CMakeLists.txt	2019-02-11 19:29:09.215805630 +0100
@@ -207,10 +207,6 @@
       )
 endif()
 
-if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-  set(bits32_veto dataframe/*.C graphs/timeSeriesFrom*.C)
-endif()
-
 #---These ones are disabled !!! ------------------------------------
 set(extra_veto
   htmlex.C
@@ -262,7 +258,6 @@
              ${classic_veto}
              ${pythia_veto}
              ${root7_veto}
-             ${bits32_veto}
              ${xrootd_veto}
              )
 
@@ -418,13 +413,6 @@
 
   list(REMOVE_ITEM pytutorials ${pyveto})
 
-  if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-    set(bits32_veto_py dataframe/*.py)
-    file(GLOB bits32_veto_py RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} dataframe/*.py)
-    list(REMOVE_ITEM pytutorials ${bits32_veto_py})
-    list(REMOVE_ITEM pytutorials pyroot/pyroot002_TTreeAsMatrix.py)
-  endif()
-
   #---Python tutorials dependencies--------------------------------------
   set(pyroot-ntuple1-depends tutorial-pyroot-hsimple-py)
   set(pyroot-h1draw-depends tutorial-pyroot-hsimple-py)
