diff -ur root-5.32.00.orig/config/Makefile.depend root-5.32.00/config/Makefile.depend
--- root-5.32.00.orig/config/Makefile.depend	2011-12-02 12:42:19.000000000 +0100
+++ root-5.32.00/config/Makefile.depend	2012-02-10 19:42:24.408218116 +0100
@@ -100,7 +100,7 @@
 CINTEXLIBDEPM          = $(REFLEXLIB)
 REFLEXDICTLIBDEPM      = $(REFLEXLIB)
 RAUTHLIBDEPM           = $(NETLIB) $(IOLIB)
-GLBSAUTHLIBDEPM        = $(RAUTHLIB)
+GLBSAUTHLIBDEPM        = $(RAUTHLIB) $(NETLIB)
 KRB5AUTHLIBDEPM        = $(RAUTHLIB) $(NETLIB)
 SRVAUTHLIBDEPM         = $(NETLIB)
 QTGSILIBDEPM           = $(GUILIB) $(GPADLIB)
@@ -366,7 +366,7 @@
 CINTEXLIBEXTRA          = lib/libReflex.lib
 REFLEXDICTLIBEXTRA      = lib/libReflex.lib
 RAUTHLIBEXTRA           = lib/libNet.lib lib/libRIO.lib
-GLBSAUTHLIBEXTRA        = lib/libRootAuth.lib
+GLBSAUTHLIBEXTRA        = lib/libRootAuth.lib lib/libNet.lib
 KRB5AUTHLIBEXTRA        = lib/libRootAuth.lib lib/libNet.lib
 SRVAUTHLIBEXTRA         = lib/libNet.lib
 QTGSILIBEXTRA           = lib/libGui.lib lib/libGpad.lib
@@ -498,7 +498,7 @@
 CINTEXLIBEXTRA          = -Llib -lReflex
 REFLEXDICTLIBEXTRA      = -Llib -lReflex
 RAUTHLIBEXTRA           = -Llib -lNet -lRIO
-GLBSAUTHLIBEXTRA        = -Llib -lRootAuth
+GLBSAUTHLIBEXTRA        = -Llib -lRootAuth -lNet
 KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet
 SRVAUTHLIBEXTRA         = -Llib -lNet
 QTGSILIBEXTRA           = -Llib -lGui -lGpad
