diff -ur root-6.06.02.orig/build/unix/compiledata.sh root-6.06.02/build/unix/compiledata.sh
--- root-6.06.02.orig/build/unix/compiledata.sh	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/build/unix/compiledata.sh	2016-03-18 16:27:27.360982151 +0100
@@ -75,7 +75,7 @@
 
 # Remove the flags turning warnings into errors or extending
 # the number of warnings.
-CXXFLAGS=`echo $CXXFLAGS | sed -e 's/-Werror //g' -e 's/-Wall //g' -e 's/-Wshadow //g'  `
+CXXFLAGS=`echo $CXXFLAGS | sed -e 's/-Werror //g' -e 's/-Werror=\S* //g' -e 's/-Wall //g' -e 's/-Wshadow //g'`
 
 # Determine the compiler version
 BXX="`basename $CXX`"
