diff -ur root-5.34.19.orig/bindings/pyroot/Module.mk root-5.34.19/bindings/pyroot/Module.mk
--- root-5.34.19.orig/bindings/pyroot/Module.mk	2014-07-09 16:50:16.000000000 +0200
+++ root-5.34.19/bindings/pyroot/Module.mk	2014-07-18 16:26:11.682503438 +0200
@@ -123,14 +123,15 @@
 $(PYTHON64O): CFLAGS += $(PYTHONINCDIR:%=-I%)
 ifeq ($(GCC_MAJOR),4)
 $(PYROOTO): CXXFLAGS += -fno-strict-aliasing
-endif
-ifneq ($(CLANG_MAJOR)$(GCC_MAJOR),)
-# Building with clang or GCC
+ifeq ($(subst $(GCC_MINOR),,0 1),0 1)
+# GCC >= 4.2
 $(PYROOTO) $(PYTHON64O) $(PYROOTDO): CXXFLAGS += -Wno-error=format
 endif
+endif
 
 ifneq ($(CLANG_MAJOR),)
 # Building with clang 
 $(PYROOTO) $(PYTHON64O) $(PYROOTDO): CXXFLAGS += -Wno-ignored-attributes
+$(PYROOTO) $(PYTHON64O) $(PYROOTDO): CXXFLAGS += -Wno-error=format
 endif
 
diff -ur root-5.34.19.orig/io/dcache/Module.mk root-5.34.19/io/dcache/Module.mk
--- root-5.34.19.orig/io/dcache/Module.mk	2014-07-09 16:50:16.000000000 +0200
+++ root-5.34.19/io/dcache/Module.mk	2014-07-18 14:41:34.700841953 +0200
@@ -68,7 +68,10 @@
 
 ##### extra rules ######
 $(DCACHEO) $(DCACHEDO): CXXFLAGS := $(filter-out -Wall,$(CXXFLAGS)) $(DCAPINCDIR:%=-I%)
-ifneq ($(CXX:g++=),$(CXX))
+ifeq ($(GCC_MAJOR),4)
+ifeq ($(subst $(GCC_MINOR),,0 1 2),0 1 2)
+# GCC >= 4.3
    $(DCACHEO) $(DCACHEDO):  CXXFLAGS += -Wno-ignored-qualifiers
 endif
+endif
 
