diff -ur root-5.26.00e.orig/gui/gui/src/TGComboBox.cxx root-5.26.00e/gui/gui/src/TGComboBox.cxx
--- root-5.26.00e.orig/gui/gui/src/TGComboBox.cxx	2010-10-13 12:39:55.000000000 +0200
+++ root-5.26.00e/gui/gui/src/TGComboBox.cxx	2010-11-12 19:21:44.043723000 +0100
@@ -765,5 +765,6 @@
    // Text font combo box dtor.
 
    for (int i = 0; i < kMaxFonts && fFonts[i] != 0; i++)
-      gVirtualX->DeleteFont(fFonts[i]);
+      if (fFonts[i] != TGTextLBEntry::GetDefaultFontStruct())
+         gVirtualX->DeleteFont(fFonts[i]);
 }
