diff -ur root-5.24.00b.orig/configure root-5.24.00b/configure
--- root-5.24.00b.orig/configure	2009-10-11 22:40:04.000000000 +0200
+++ root-5.24.00b/configure	2009-11-26 09:57:20.562269489 +0100
@@ -2916,9 +2916,8 @@
           $cwhich afterimage-config
           # we found afterimage-config - now check the version number
           asimage_version=`afterimage-config --version | awk 'BEGIN { FS="." } { printf "%d", $1 * 1000 + $2 }'`
-          #if test $asimage_version -gt 1003 ; then
-          if test $asimage_version -gt 0 ; then
-             result "WARNING: System libAfterImage is too new, using built-in"
+          if test $asimage_version -lt 1020 ; then
+             result "WARNING: System libAfterImage is too old, using built-in"
           else
              enable_builtin_afterimage="no"
              asextralib=`afterimage-config --libs`
