diff -ur root-6.06.02.orig/hist/hist/src/TEfficiency.cxx root-6.06.02/hist/hist/src/TEfficiency.cxx
--- root-6.06.02.orig/hist/hist/src/TEfficiency.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/hist/hist/src/TEfficiency.cxx	2016-03-08 21:34:59.673902466 +0100
@@ -329,7 +329,7 @@
 parameters are shown in the plot below.
 
 Begin_Macro(source)
-   {
+{
       //canvas only needed for the documentation
       TCanvas* c1 = new TCanvas("c1","",600,400);
       c1->SetFillStyle(1001);
@@ -363,7 +363,7 @@
 
       //only for this documentation
       return c1;
-   }
+}
 End_Macro
 
 
diff -ur root-6.06.02.orig/hist/hist/src/TH1.cxx root-6.06.02/hist/hist/src/TH1.cxx
--- root-6.06.02.orig/hist/hist/src/TH1.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/hist/hist/src/TH1.cxx	2016-03-08 13:45:25.689411795 +0100
@@ -1809,9 +1809,9 @@
 /// of comparison of the unweighted histogram with 217 events (minimal expected
 /// frequency equal to one) and the weighted histogram with 500 events (minimal
 /// expected frequency equal to 25)
-///Begin_Macro
+/// Begin_Macro
 /// ../../../tutorials/math/chi2test.C(17)
-///End_Macro
+/// End_Macro
 /// Fig 2. An example of comparison of the unweighted histogram with 217 events
 /// and the weighted histogram with 500 events:
 ///   a) unweighted histogram;
