diff --git a/bindings/pyroot/pythonizations/python/ROOT/__init__.py b/bindings/pyroot/pythonizations/python/ROOT/__init__.py
index 227f80765b..eb03251c28 100644
--- a/bindings/pyroot/pythonizations/python/ROOT/__init__.py
+++ b/bindings/pyroot/pythonizations/python/ROOT/__init__.py
@@ -25,6 +25,12 @@ environ['CPPYY_API_PATH'] = 'none'
 # Prevent cppyy from filtering ROOT libraries
 environ['CPPYY_NO_ROOT_FILTER'] = '1'
 
+try:
+    from ctypes import cdll
+    libCore = cdll.LoadLibrary("libCore.so")
+except:
+    print("An exception occurred while loading libCore.so with ctypes")
+
 import cppyy
 
 # import libROOTPythonizations with Python version number
