diff -ur root-6.06.02.orig/gui/ged/src/TArrowEditor.cxx root-6.06.02/gui/ged/src/TArrowEditor.cxx
--- root-6.06.02.orig/gui/ged/src/TArrowEditor.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/gui/ged/src/TArrowEditor.cxx	2016-03-29 15:00:55.706652094 +0200
@@ -30,7 +30,7 @@
 shape, size, angle.</h2></center>
 The picture below shows this interface.
 End_Html
-Begin_Macro(GUI)
+~~~
 {
    gROOT->ProcessLine(".x $ROOTSYS/tutorials/graphics/arrow.C");
    TCanvas *c1 = (TCanvas *)gROOT->GetListOfCanvases()->FindObject("c1");
@@ -41,7 +41,8 @@
    r->SetWindowSize(700,500);
    return r->GetMainFrame();
 }
-End_Macro */
+~~~
+*/
 
 
 ////////////////////////////////////////////////////////////////////////////////
diff -ur root-6.06.02.orig/gui/gui/src/TGTable.cxx root-6.06.02/gui/gui/src/TGTable.cxx
--- root-6.06.02.orig/gui/gui/src/TGTable.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/gui/gui/src/TGTable.cxx	2016-03-29 15:05:17.952862862 +0200
@@ -72,7 +72,7 @@
 <br><br>
 A simple macro to use a TGTable with a TGSimpleTableInterface:
 End_Html
-Begin_Macro(source, gui)
+~~~
 {
    Int_t i = 0, j = 0;
    UInt_t nrows = 6, ncolumns = 5;
@@ -111,13 +111,13 @@
 
    return mainframe;
 }
-End_Macro
+~~~
 Begin_Html
 
 It is also possible to visualise data from a tree. A simple macro
 showing the use of a TTreeTableInterface follows.
 End_Html
-Begin_Macro(source, gui)
+~~~
 {
    // Open a root file.
    TFile *file = new TFile("$ROOTSYS/tutorials/hsimple.root");
@@ -153,7 +153,7 @@
 
    return mainframe;
 }
-End_Macro
+~~~
 */
 
 // const TGGC *TGTable::fgDefaultSelectGC = 0;
