diff -ur root-6.06.02.orig/math/physics/inc/TVector2.h root-6.06.02/math/physics/inc/TVector2.h
--- root-6.06.02.orig/math/physics/inc/TVector2.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/physics/inc/TVector2.h	2016-03-06 19:07:23.115476536 +0100
@@ -96,7 +96,8 @@
    TVector2 Rotate (Double_t phi) const;
 
                                         // returns phi angle in the interval [0,2*PI)
-   static Double_t Phi_0_2pi(Double_t x);                                                                               // returns phi angle in the interval
+   static Double_t Phi_0_2pi(Double_t x);
+
                                         // returns phi angle in the interval [-PI,PI)
    static Double_t Phi_mpi_pi(Double_t x);
 
