From f33ac48317fc1a144d6a7f0a7028e898b664e3a2 Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Thu, 28 Jun 2018 23:02:46 +0200
Subject: [PATCH] Add missing include for std::shuffle

---
 tmva/tmva/test/TestRandomGenerator.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tmva/tmva/test/TestRandomGenerator.cxx b/tmva/tmva/test/TestRandomGenerator.cxx
index c8f0f50ee5f..02f941d7f26 100644
--- a/tmva/tmva/test/TestRandomGenerator.cxx
+++ b/tmva/tmva/test/TestRandomGenerator.cxx
@@ -10,6 +10,7 @@
 #include "TMVA/Tools.h"
 
 // Stdlib
+#include <algorithm>
 #include <iostream>
 
 // External
