diff -ur root-6.20.02.orig/bindings/pyroot/JsMVA/JPyInterface.py root-6.20.02/bindings/pyroot/JsMVA/JPyInterface.py
--- root-6.20.02.orig/bindings/pyroot/JsMVA/JPyInterface.py	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/bindings/pyroot/JsMVA/JPyInterface.py	2020-03-15 20:07:18.921423559 +0100
@@ -188,7 +188,7 @@
 ## Class for creating the output scripts and inserting them to cell output
 class JsDraw:
     ## Base repository
-    __jsMVARepo = "https://root.cern.ch/js/jsmva/latest"
+    __jsMVARepo = "/static/JsMVA"
  
     ## String containing the link to JavaScript files
     __jsMVASourceDir = __jsMVARepo + "/js"
diff -ur root-6.20.02.orig/bindings/pyroot/JupyROOT/helpers/utils.py root-6.20.02/bindings/pyroot/JupyROOT/helpers/utils.py
--- root-6.20.02.orig/bindings/pyroot/JupyROOT/helpers/utils.py	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/bindings/pyroot/JupyROOT/helpers/utils.py	2020-03-15 20:07:18.921423559 +0100
@@ -82,7 +82,7 @@
 function requirejs_success(base_url) {{
     return function() {{
         require.config({{
-            baseUrl: base_url + 'static/'
+            baseUrl: base_url + 'static/jsroot/'
         }});
         display_{jsDivId}();
     }}
diff -ur root-6.20.02.orig/etc/notebook/JsMVA/js/JsMVA.js root-6.20.02/etc/notebook/JsMVA/js/JsMVA.js
--- root-6.20.02.orig/etc/notebook/JsMVA/js/JsMVA.js	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/etc/notebook/JsMVA/js/JsMVA.js	2020-03-15 20:07:18.922423549 +0100
@@ -16,7 +16,7 @@
 
 (function(factory){
 
-    var JSROOT_source_dir = "https://root.cern.ch/js/notebook/scripts/";
+    var JSROOT_source_dir = "/static/jsroot/scripts/";
 
     var url = "";
     if (requirejs.s.contexts.hasOwnProperty("_")) {
diff -ur root-6.20.02.orig/etc/notebook/JsMVA/js/NetworkDesigner.js root-6.20.02/etc/notebook/JsMVA/js/NetworkDesigner.js
--- root-6.20.02.orig/etc/notebook/JsMVA/js/NetworkDesigner.js	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/etc/notebook/JsMVA/js/NetworkDesigner.js	2020-03-15 20:07:18.922423549 +0100
@@ -19,7 +19,7 @@
         paths: {
             "jquery-connections": baseURL + "jquery.connections.min",
             "jquery-timing": baseURL + "jquery-timing.min",
-            "d3": "/static/scripts/d3.min"
+            "d3": "/static/jsroot/scripts/d3.min"
         },
         shim: {
             "jquery-ui": {
