diff -ur root-6.06.02.orig/cmake/modules/SetUpLinux.cmake root-6.06.02/cmake/modules/SetUpLinux.cmake
--- root-6.06.02.orig/cmake/modules/SetUpLinux.cmake	2016-04-06 09:42:10.258085039 +0200
+++ root-6.06.02/cmake/modules/SetUpLinux.cmake	2016-04-06 10:07:33.109883779 +0200
@@ -19,7 +19,7 @@
     message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
   endif()
 elseif(${SYSCTL_OUTPUT} MATCHES aarch64)
-  message(STATUS "Found a 64bit system")
+  message(STATUS "Found a 64bit ARM system")
   set(SPECIAL_CINT_FLAGS "-DG__64BIT")
   if(CMAKE_COMPILER_IS_GNUCXX)
     message(STATUS "Found GNU compiler collection")
@@ -42,6 +42,18 @@
   else()
     message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
   endif()
+elseif(${SYSCTL_OUTPUT} MATCHES arm)
+  message(STATUS "Found a 32bit ARM system")
+  set(SPECIAL_CINT_FLAGS "")
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    message(STATUS "Found GNU compiler collection")
+    set(ROOT_ARCHITECTURE linuxarm)
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+    message(STATUS "Found CLANG compiler")
+    set(ROOT_ARCHITECTURE linuxarm)
+  else()
+    message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
+  endif()
 else()
   message(STATUS "Found a 32bit system")
   set(BIT_ENVIRONMENT "-m32")
diff -ur root-6.06.02.orig/core/pcre/CMakeLists.txt root-6.06.02/core/pcre/CMakeLists.txt
--- root-6.06.02.orig/core/pcre/CMakeLists.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/pcre/CMakeLists.txt	2016-04-06 09:53:37.247748077 +0200
@@ -52,6 +52,8 @@
     set(PCRE_CFLAGS "-m32 -O")
   elseif(ROOT_ARCHITECTURE MATCHES linuxarm64)
     set(PCRE_CFLAGS "-O")
+  elseif(ROOT_ARCHITECTURE MATCHES linuxarm)
+    set(PCRE_CFLAGS "-O")
   elseif(ROOT_ARCHITECTURE MATCHES linux)
     set(PCRE_CFLAGS "-m32 -O")
   elseif(ROOT_ARCHITECTURE MATCHES macosx64)
diff -ur root-6.06.02.orig/graf2d/asimage/BuildAfterImage.cmake root-6.06.02/graf2d/asimage/BuildAfterImage.cmake
--- root-6.06.02.orig/graf2d/asimage/BuildAfterImage.cmake	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf2d/asimage/BuildAfterImage.cmake	2016-04-06 09:53:37.258748191 +0200
@@ -47,6 +47,8 @@
     set(AFTER_CC "${AFTER_CC} -m64")
   elseif(ROOT_ARCHITECTURE MATCHES linuxarm64)
     set(AFTER_CC "${AFTER_CC}")
+  elseif(ROOT_ARCHITECTURE MATCHES linuxarm)
+    set(AFTER_CC "${AFTER_CC}")
   elseif(ROOT_ARCHITECTURE MATCHES linux)
     set(AFTER_CC "${AFTER_CC} -m32")
   elseif(ROOT_ARCHITECTURE MATCHES macosx64)
diff -ur root-6.06.02.orig/graf2d/freetype/CMakeLists.txt root-6.06.02/graf2d/freetype/CMakeLists.txt
--- root-6.06.02.orig/graf2d/freetype/CMakeLists.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf2d/freetype/CMakeLists.txt	2016-04-06 09:59:35.237876779 +0200
@@ -52,6 +52,10 @@
     set(FREETYPE_CFLAGS "-m64 -O")
   elseif(ROOT_ARCHITECTURE MATCHES linuxppc64gcc)
     set(FREETYPE_CFLAGS "-m64 -O")
+  elseif(ROOT_ARCHITECTURE MATCHES linuxarm64)
+    set(FREETYPE_CFLAGS "-O")
+  elseif(ROOT_ARCHITECTURE MATCHES linuxarm)
+    set(FREETYPE_CFLAGS "-O")
   elseif(ROOT_ARCHITECTURE MATCHES linux)
     set(FREETYPE_CFLAGS "-m32 -O")
   elseif(ROOT_ARCHITECTURE MATCHES macosx64)
diff -ur root-6.06.02.orig/interpreter/CMakeLists.txt root-6.06.02/interpreter/CMakeLists.txt
--- root-6.06.02.orig/interpreter/CMakeLists.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/interpreter/CMakeLists.txt	2016-04-06 09:53:37.270748316 +0200
@@ -17,6 +17,8 @@
 
 if(ROOT_ARCHITECTURE MATCHES linuxarm64)
   set(ROOT_CLING_TARGET "AArch64")
+elseif(ROOT_ARCHITECTURE MATCHES linuxarm)
+  set(ROOT_CLING_TARGET "ARM")
 elseif(ROOT_ARCHITECTURE MATCHES linuxppc64gcc)
   set(ROOT_CLING_TARGET "PowerPC")
 elseif(ROOT_ARCHITECTURE MATCHES linux)
