diff -ur root-6.06.02.orig/configure root-6.06.02/configure
--- root-6.06.02.orig/configure	2016-03-20 15:54:26.038192410 +0100
+++ root-6.06.02/configure	2016-03-20 16:48:49.767492974 +0100
@@ -202,7 +202,7 @@
     enable_unuran=yes
     enable_vc=yes
     enable_vdt=yes
-    #enable_r=yes
+    enable_r=yes
 fi
 
 # Remove old log file
@@ -2288,7 +2288,7 @@
         f=`echo $1 | sed -e 's/--//' -e 's/-/_/g' -e 's/++11/xx11/'`
         eval prev='$'`echo ${f}`
         eval prev_expl='$'`echo ${f}_explicit`
-        if test ! "x`echo ${deprecated_options} | grep ${f}`" = "x"; then
+        if test ! "x`echo ${deprecated_options}' ' | grep ${f}' '`" = "x"; then
             echo "WARNING: option $1 is deprecated and ignored"'!'
         else
             if test "${prev}:${prev_expl}" = "yes:"; then
@@ -2318,7 +2318,7 @@
         f=`echo $1 | sed -e 's/--disable/enable/' -e 's/-/_/g' -e 's/++11/xx11/'`
         eval prev='$'`echo ${f}`
         eval prev_expl='$'`echo ${f}_explicit`
-        if test ! "x`echo ${deprecated_options} | grep ${f}`" = "x"; then
+        if test ! "x`echo ${deprecated_options}' ' | grep ${f}' '`" = "x"; then
             echo "WARNING: option $1 is deprecated and ignored"'!'
         else
             if test "${prev}:${prev_expl}" = "xno:"; then
