diff -ur root-6.12.04.orig/tutorials/pythia/pythia8.C root-6.12.04/tutorials/pythia/pythia8.C
--- root-6.12.04.orig/tutorials/pythia/pythia8.C	2017-12-13 08:27:42.000000000 +0100
+++ root-6.12.04/tutorials/pythia/pythia8.C	2018-01-10 07:01:23.215029332 +0100
@@ -8,11 +8,6 @@
 ///  root > .x pythia8.C
 /// ~~~
 ///
-/// Note that before executing this script,
-///
-///  - the env variable PYTHIA8 must point to the pythia8100 (or newer) directory
-///  - the env variable PYTHIA8DATA must be defined and it must point to $PYTHIA8/xmldoc
-///
 /// \macro_code
 ///
 /// \author Andreas Morsch
@@ -27,26 +22,6 @@
 
 void pythia8(Int_t nev  = 100, Int_t ndeb = 1)
 {
-   const char *p8dataenv = gSystem->Getenv("PYTHIA8DATA");
-   if (!p8dataenv) {
-      const char *p8env = gSystem->Getenv("PYTHIA8");
-      if (!p8env) {
-         Error("pythia8.C",
-               "Environment variable PYTHIA8 must contain path to pythia directory!");
-         return;
-      }
-      TString p8d = p8env;
-      p8d += "/xmldoc";
-      gSystem->Setenv("PYTHIA8DATA", p8d);
-   }
-
-   const char* path = gSystem->ExpandPathName("$PYTHIA8DATA");
-   if (gSystem->AccessPathName(path)) {
-         Error("pythia8.C",
-               "Environment variable PYTHIA8DATA must contain path to $PYTHIA8/xmldoc directory !");
-      return;
-   }
-
 // Load libraries
    gSystem->Load("libEG");
    gSystem->Load("libEGPythia8");
