diff -ur root-6.06.02.orig/core/utils/CMakeLists.txt root-6.06.02/core/utils/CMakeLists.txt
--- root-6.06.02.orig/core/utils/CMakeLists.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/utils/CMakeLists.txt	2016-03-18 00:28:13.225932434 +0100
@@ -36,7 +36,9 @@
                                     ${CLING_LIBRARIES} ${CMAKE_DL_LIBS} ${CMAKE_THREAD_LIBS_INIT}
                                     ${corelinklibs})
 
-add_dependencies(rootcling CLING LLVMRES)
+# The dependency on Cling was added to prevent Cling (libCling) and
+# rootcling from being linked in parallel.
+add_dependencies(rootcling CLING LLVMRES Cling)
 
 ROOT_EXECUTABLE(rlibmap src/rlibmap.cxx)
 
diff -ur root-6.06.02.orig/core/utils/Module.mk root-6.06.02/core/utils/Module.mk
--- root-6.06.02.orig/core/utils/Module.mk	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/utils/Module.mk	2016-03-18 00:10:57.995028698 +0100
@@ -25,9 +25,11 @@
 
 CLINGMETAUTILSO    = $(METAUTILSTO) $(METAUTILSOLLVM)
 ROOTCLINGEXEEXTRAO = $(COREO) $(COREDO) $(IOO) $(IODO) $(THREADO) $(THREADDO) $(METAOLLVM)
+# The dependency on $(CLINGLIB) was added to prevent $(CLINGLIB) and
+# $(ROOTCLINGEXE) from being linked in parallel.
 $(ROOTCLINGEXE): $(ROOTCLINGO) $(ROOTCLINGUTILO) $(ROOTCLINGTCLINGO) \
 	   $(CLINGMETAUTILSO) $(SNPRINTFO) $(CLINGO) $(ROOTCLINGEXEEXTRAO) \
-           $(PCREDEP) $(CORELIBDEP)
+           $(PCREDEP) $(CORELIBDEP) $(CLINGLIB)
 	$(LD) $(LDFLAGS) $(OSTHREADLIBDIR) $(OSTHREADLIB) -o $@ $(ROOTCLINGO) $(ROOTCLINGUTILO) \
 	   $(ROOTCLINGTCLINGO) $(CLINGMETAUTILSO) \
 	   $(SNPRINTFO)  $(CLINGO) $(ROOTCLINGEXEEXTRAO) $(CLINGLIBEXTRA) \
