diff -ur root-6.20.02.orig/bindings/pyroot/ROOT.py root-6.20.02/bindings/pyroot/ROOT.py
--- root-6.20.02.orig/bindings/pyroot/ROOT.py	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/bindings/pyroot/ROOT.py	2020-03-15 19:50:25.303179253 +0100
@@ -438,9 +438,7 @@
 # This function is injected as method to the respective classes in Pythonize.cxx.
 _root._RDataFrameAsNumpy = _RDataFrameAsNumpy
 
-# This Pythonisation is there only for 64 bits builds
-if (sys.maxsize > 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-    _root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
+_root.CreateScopeProxy( "TTree" ).AsMatrix = _TTreeAsMatrix
 
 
 ### RINT command emulation ------------------------------------------------------
diff -ur root-6.20.02.orig/build/unix/makepchinput.py root-6.20.02/build/unix/makepchinput.py
--- root-6.20.02.orig/build/unix/makepchinput.py	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/build/unix/makepchinput.py	2020-03-15 19:50:25.304179242 +0100
@@ -252,9 +252,6 @@
                            "math/vdt",
                            "tmva/rmva"]
 
-   if (sys.platform != 'win32' and sys.maxsize <= 2**32): # https://docs.python.org/3/library/platform.html#cross-platform
-      PCHPatternsBlacklist.append("tree/dataframe")
-
    accepted = isAnyPatternInString(PCHPatternsWhitelist,dirName) and \
                not isAnyPatternInString(PCHPatternsBlacklist,dirName)
 
diff -ur root-6.20.02.orig/tree/dataframe/test/dataframe_cache.cxx root-6.20.02/tree/dataframe/test/dataframe_cache.cxx
--- root-6.20.02.orig/tree/dataframe/test/dataframe_cache.cxx	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tree/dataframe/test/dataframe_cache.cxx	2020-03-15 19:50:09.075351525 +0100
@@ -224,8 +224,6 @@
 
 }
 
-#ifdef R__B64
-
 TEST(Cache, Regex)
 {
 
@@ -334,5 +332,3 @@
 
    gSystem->Unlink(fileName);
 }
-
-#endif // R__B64
diff -ur root-6.20.02.orig/tree/dataframe/test/datasource_arrow.cxx root-6.20.02/tree/dataframe/test/datasource_arrow.cxx
--- root-6.20.02.orig/tree/dataframe/test/datasource_arrow.cxx	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tree/dataframe/test/datasource_arrow.cxx	2020-03-15 19:50:09.075351525 +0100
@@ -163,8 +163,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(RArrowDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RArrowDS(createTestTable(), {}));
@@ -236,5 +234,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.20.02.orig/tree/dataframe/test/datasource_csv.cxx root-6.20.02/tree/dataframe/test/datasource_csv.cxx
--- root-6.20.02.orig/tree/dataframe/test/datasource_csv.cxx	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tree/dataframe/test/datasource_csv.cxx	2020-03-15 19:57:05.490930922 +0100
@@ -195,8 +195,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(RCsvDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RCsvDS(fileName0));
@@ -292,5 +290,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.20.02.orig/tree/dataframe/test/datasource_root.cxx root-6.20.02/tree/dataframe/test/datasource_root.cxx
--- root-6.20.02.orig/tree/dataframe/test/datasource_root.cxx	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tree/dataframe/test/datasource_root.cxx	2020-03-15 19:50:09.076351514 +0100
@@ -117,8 +117,6 @@
 }
 #endif
 
-#ifdef R__B64
-
 TEST(TRootTDS, FromARDF)
 {
    std::unique_ptr<RDataSource> tds(new RRootDS(treeName, fileGlob));
@@ -190,5 +188,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.20.02.orig/tree/dataframe/test/datasource_trivial.cxx root-6.20.02/tree/dataframe/test/datasource_trivial.cxx
--- root-6.20.02.orig/tree/dataframe/test/datasource_trivial.cxx	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tree/dataframe/test/datasource_trivial.cxx	2020-03-15 19:50:09.076351514 +0100
@@ -132,8 +132,6 @@
    EXPECT_EQ(*tdfAll.Count(), 20ULL);
 }
 
-#ifdef R__B64
-
 TEST(RTrivialDS, FromARDFWithJitting)
 {
    std::unique_ptr<RDataSource> tds(new RTrivialDS(32));
@@ -236,5 +234,3 @@
 }
 
 #endif // R__USE_IMT
-
-#endif // R__B64
diff -ur root-6.20.02.orig/tutorials/CMakeLists.txt root-6.20.02/tutorials/CMakeLists.txt
--- root-6.20.02.orig/tutorials/CMakeLists.txt	2020-03-15 16:21:25.000000000 +0100
+++ root-6.20.02/tutorials/CMakeLists.txt	2020-03-15 19:59:35.808335176 +0100
@@ -255,10 +255,6 @@
   set(root7_veto v7/ntuple/ntpl004_dimuon.C)
 endif()
 
-if( CMAKE_SIZEOF_VOID_P EQUAL 4 )
-  set(bits32_veto dataframe/*.C graphs/timeSeriesFrom*.C v7/ntuple/ntpl004_dimuon.C)
-endif()
-
 #---These ones are disabled !!! ------------------------------------
 set(extra_veto
   legacy/benchmarks.C
@@ -311,7 +307,6 @@
              ${classic_veto}
              ${pythia_veto}
              ${root7_veto}
-             ${bits32_veto}
              ${xrootd_veto}
              ${mlp_veto}
              ${spectrum_veto}
@@ -513,7 +508,6 @@
   list(REMOVE_ITEM pytutorials ${pyveto})
 
   if(NOT dataframe)
-    set(dataframe_veto_py dataframe/*.py)
     file(GLOB dataframe_veto_py RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} dataframe/*.py tmva/tmva*.py)
     list(REMOVE_ITEM pytutorials ${dataframe_veto_py})
     list(REMOVE_ITEM pytutorials pyroot/pyroot002_TTreeAsMatrix.py)
