--- root-5.26.00d.orig/tutorials/tree/cernbuild.C	2010-08-29 10:20:56.069801363 +0200
+++ root-5.26.00d/tutorials/tree/cernbuild.C	2010-08-29 15:52:51.000000000 +0200
@@ -22,7 +22,7 @@
    TString dir = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
    dir.ReplaceAll("cernbuild.C","");
    dir.ReplaceAll("/./","/");
-   FILE *fp = fopen(Form("%scernstaff.dat",dir.Data()),"r");
+   FILE *fp = fopen((dir+"cernstaff.dat").Data(),"r");
 
    TFile *hfile = 0;
    if (get) {
