diff -ur root-5.24.00b.orig/tree/treeviewer/src/TParallelCoord.cxx root-5.24.00b/tree/treeviewer/src/TParallelCoord.cxx
--- root-5.24.00b.orig/tree/treeviewer/src/TParallelCoord.cxx	2009-10-11 22:39:38.000000000 +0200
+++ root-5.24.00b/tree/treeviewer/src/TParallelCoord.cxx	2009-11-30 22:21:44.200770648 +0100
@@ -71,24 +71,6 @@
 <center><h2>Candle chart:</h2></center>
 <p>
 TParallelCoord can also be used to display a candle chart. In that mode, every variable is drawn in the same scale. The candle chart can be combined with the parallel coordinates mode, drawing the candle sticks over the axes.
-End_Html
-Begin_Macro(source)
-{
-   TCanvas *c1 = new TCanvas("c1");
-   TFile *f = TFile::Open("$ROOTSYS/tutorials/tree/cernstaff.root");
-   TTree *T = (TTree*)f->Get("T");
-   T->Draw("Age:Grade:Step:Cost:Division:Nation","","para");
-   TParallelCoord* para = (TParallelCoord*)gPad->GetListOfPrimitives()->FindObject("ParaCoord");
-   TParallelCoordVar* grade = (TParallelCoordVar*)para->GetVarList()->FindObject("Grade");
-   grade->AddRange(new TParallelCoordRange(grade,11.5,14));
-   para->AddSelection("less30");
-   para->GetCurrentSelection()->SetLineColor(kViolet);
-   TParallelCoordVar* age = (TParallelCoordVar*)para->GetVarList()->FindObject("Age");
-   age->AddRange(new TParallelCoordRange(age,21,30));
-   return c1;
-}
-End_Macro
-Begin_Html
 <p> Some references:
 <ul>
 <li> Alfred Inselberg's Homepage <http://www.math.tau.ac.il/~aiisreal>, with Visual Tutorial, History, Selected Publications and Applications.</li>
