diff -ur root-5.34.32.orig/io/gfal/src/TGFALFile.cxx root-5.34.32/io/gfal/src/TGFALFile.cxx
--- root-5.34.32.orig/io/gfal/src/TGFALFile.cxx	2015-06-23 17:56:20.000000000 +0200
+++ root-5.34.32/io/gfal/src/TGFALFile.cxx	2015-11-17 14:33:31.069759901 +0100
@@ -68,7 +68,7 @@
 }
 
 // GFAL2 doesn't use special names for 64 bit versions
-#ifdef _GFAL2_API_
+#if defined(_GFAL2_API_) || defined(GFAL2_API_H_)
 #define gfal_lseek64   gfal_lseek
 #define gfal_open64    gfal_open
 #define gfal_readdir64 gfal_readdir
