diff -ur root-5.28.00h.orig/config/Makefile.in root-5.28.00h/config/Makefile.in
--- root-5.28.00h.orig/config/Makefile.in	2011-11-24 17:12:55.000000000 +0100
+++ root-5.28.00h/config/Makefile.in	2013-03-02 08:09:02.230002472 +0100
@@ -154,6 +154,7 @@
 GFALLIBDIR     := @gfallibdir@
 GFALCLILIB     := @gfallib@
 GFALINCDIR     := $(filter-out /usr/include, @gfalincdir@)
+SRMIFCEINCDIR  := $(filter-out /usr/include, @srmifceincdir@)
 
 BUILDGLITE     := @buildglite@
 GLITELIBDIR    := @glitelibdir@
diff -ur root-5.28.00h.orig/configure root-5.28.00h/configure
--- root-5.28.00h.orig/configure	2011-11-24 17:12:57.000000000 +0100
+++ root-5.28.00h/configure	2013-03-02 08:10:33.684863607 +0100
@@ -1426,6 +1426,7 @@
   gviz-incdir        Graphviz support, location of gvc.h
   gviz-libdir        Graphviz support, location of libgvplugin_core
   gfal-incdir        GFAL support, location of gfal_api.h
+  srm-ifce-incdir    GFAL support, location of gfal_srm_ifce_types.h
   gfal-libdir        GFAL support, location of libgfal
   glew-incdir        GLEW support, location of glew.h
   glew-libdir        GLEW support, location of libGLEW
@@ -1733,6 +1734,7 @@
       --with-gviz-libdir=*)    gvizlibdir=$optarg    ; enable_gviz="yes"    ;;
       --with-gfal-incdir=*)    gfalincdir=$optarg    ; enable_gfal="yes"    ;;
       --with-gfal-libdir=*)    gfallibdir=$optarg    ; enable_gfal="yes"    ;;
+      --with-srm-ifce-incdir=*) srmifceincdir=$optarg;;
       --with-glew-incdir=*)    glewincdir=$optarg    ; enable_builtin_glew=no;;
       --with-glew-libdir=*)    glewlibdir=$optarg    ; enable_builtin_glew=no;;
       --with-gsl-incdir=*)     gslincdir=$optarg     ; enable_gsl="yes"     ;;
@@ -3199,6 +3201,12 @@
     gfallib=$found_lib
     gfallibdir=$found_dir
 
+    check_header "gfal_srm_ifce_types.h" "$srmifceincdir" \
+        /opt/lcg/include /opt/srm-ifce/include /usr/local/lcg/include \
+        /usr/local/srm-ifce/include
+    srmifceinc=$found_hdr
+    srmifceincdir=$found_dir
+
     if test "x$gfalincdir" = "x" || test "x$gfallib" = "x"; then
         enable_gfal="no"
     fi
@@ -6341,6 +6349,7 @@
     -e "s|@exceptions@|$enable_exceptions|"     \
     -e "s|@explicitlink@|$enable_explicitlink|" \
     -e "s|@gfalincdir@|$gfalincdir|"            \
+    -e "s|@srmifceincdir@|$srmifceincdir|"      \
     -e "s|@gfallib@|$gfallib|"                  \
     -e "s|@gfallibdir@|$gfallibdir|"            \
     -e "s|@enable_xft@|$enable_xft|"            \
diff -ur root-5.28.00h.orig/io/gfal/Module.mk root-5.28.00h/io/gfal/Module.mk
--- root-5.28.00h.orig/io/gfal/Module.mk	2011-11-24 17:12:56.000000000 +0100
+++ root-5.28.00h/io/gfal/Module.mk	2013-03-02 08:04:00.349761906 +0100
@@ -70,3 +70,6 @@
 ##### extra rules ######
 $(GFALO) $(GFALDO): CXXFLAGS := $(filter-out -Wshadow,$(CXXFLAGS))
 $(GFALO): CXXFLAGS += $(GFALINCDIR:%=-I%)
+$(GFALO): CXXFLAGS += $(SRMIFCEINCDIR:%=-I%)
+$(GFALO): CXXFLAGS += $(shell pkg-config --cflags glib-2.0)
+$(GFALO): CXXFLAGS += -D_FILE_OFFSET_BITS=64
