diff --git a/documentation/doxygen/Makefile b/documentation/doxygen/Makefile
index 4e6b47728d..a886c1b729 100644
--- a/documentation/doxygen/Makefile
+++ b/documentation/doxygen/Makefile
@@ -8,7 +8,6 @@ export DOXYGEN_ROOT_VERSION := $(shell root -l -b -q -e 'printf("%s\n", gROOT->G
 export DOXYGEN_EXAMPLE_PATH := $(DOXYGEN_OUTPUT_DIRECTORY)/macros
 export DOXYGEN_IMAGE_PATH := $(DOXYGEN_OUTPUT_DIRECTORY)/html
 export DOXYGEN_NOTEBOOK_PATH := $(DOXYGEN_OUTPUT_DIRECTORY)/notebooks
-export DOXYGEN_NOTEBOOK_PATH_PARALLEL := $(DOXYGEN_NOTEBOOK_PATH)/parallel
 export DOXYGEN_STRIP_PATH := $(shell cd $(PWD)/../..; pwd)
 export DOXYGEN_INCLUDE_PATH := $(shell find $(DOXYGEN_STRIP_PATH) -type d -name dictpch -prune -o -type d -name inc)
 
@@ -33,14 +32,11 @@ doxygen: filter
 	$(call MkDir,$(DOXYGEN_OUTPUT_DIRECTORY))
 	$(call MkDir,$(DOXYGEN_EXAMPLE_PATH))
 	$(call MkDir,$(DOXYGEN_NOTEBOOK_PATH))
-	$(call MkDir,$(DOXYGEN_NOTEBOOK_PATH_PARALLEL))
-	time python ./parallelNotebooks.py
 	./makehtmlfooter.sh > htmlfooter.html
 	doxygen
 	./listlibs.sh
 	rm -rf files c1* *.ps *.png *.svg *.pdf *.root *.xpm *.out *.dat *.dtd *.dot *.txt listofclass.sh
-	rm $(DOXYGEN_NOTEBOOK_PATH)/*.root
-	rm $(DOXYGEN_NOTEBOOK_PATH_PARALLEL)/*.root
+	rm -f $(DOXYGEN_NOTEBOOK_PATH)/*.root
 
 clean:
 	rm -r $(DOXYGEN_OUTPUT_DIRECTORY) filter htmlfooter.html dataset*
