diff -ur root-5.32.00.orig/configure root-5.32.00/configure
--- root-5.32.00.orig/configure	2011-12-02 12:42:22.000000000 +0100
+++ root-5.32.00/configure	2012-02-10 17:36:40.090907758 +0100
@@ -3183,7 +3183,7 @@
         fi
     fi
 
-    check_library "librfio libdpm libshift shiftmd shift" "$enable_shared" \
+    check_library "librfio libdpm libcastorrfio libshift shiftmd shift" "$enable_shared" \
         "$shiftlibdir" \
         $RFIO ${RFIO:+$RFIO/lib} /cern/pro/lib /cern/new/lib /cern/old/lib \
         /opt/shift/lib /usr/local/shift/lib /usr/lib/shift \
@@ -3194,6 +3194,10 @@
     if test "x$shiftincdir" = "x" || test "x$shiftlib" = "x"; then
         enable_rfio="no"
     else
+        case $shiftlib in
+            *dpm)        shiftlib="$shiftlib -llcgdm" ;;
+            *castorrfio) shiftlib="$shiftlib -lcastorcommon" ;;
+        esac
         case $platform in
            linux | macosx)
                 for i in "" -ladns ; do
@@ -3203,14 +3207,7 @@
                         break ;
                     fi
                 done
-                # Explicit linking: which libraries are needed to satisfy symbols
-                # from libshift?
-                # 2.1.9 has new library layout
-                shiftlibdep="-lcastorrfio -lcastorclient -lcastorcommon"
-                check_link "$shiftlib $shiftlibdep" "$shiftlibdir" rfio_fchmod
-                if test $link_result -eq 1 ; then
-                    shiftlib="$shiftlib $shiftlibdep"
-                else
+                if test $link_result -ne 1 ; then
                     enable_rfio="no"
                 fi
                 ;;
@@ -3280,7 +3277,7 @@
         fi
     fi
 
-    check_library "libshift shiftmd shift" "$enable_shared" "$castorlibdir" \
+    check_library "libcastorrfio libshift shiftmd shift" "$enable_shared" "$castorlibdir" \
         $CASTOR ${CASTOR:+$CASTOR/lib} /cern/pro/lib /cern/new/lib \
         /cern/old/lib /opt/shift/lib /usr/local/shift/lib /usr/lib/shift \
         /usr/local/lib/shift /usr/lib /usr/local/lib
@@ -3290,6 +3287,9 @@
     if test "x$castorincdir" = "x" || test "x$castorlib" = "x"; then
         enable_castor="no"
     else
+        case $castorlib in
+            *castorrfio) castorlib="$castorlib -lcastorclient -lcastorns -lcastorcommon" ;;
+        esac
         case $platform in
            linux | macosx)
                 for i in "" -ladns ; do
@@ -3299,14 +3299,7 @@
                         break ;
                     fi
                 done
-                # Explicit linking: which libraries are needed to satisfy symbols
-                # from libshift?
-                # 2.1.9 has new library layout
-                castorlibdep="-lcastorrfio -lcastorclient -lcastorns -lcastorcommon"
-                check_link "$castorlib $castorlibdep" "$castorlibdir" rfio_fchmod
-                if test $link_result -eq 1 ; then
-                    castorlib="$castorlib $castorlibdep"
-                else
+                if test $link_result -ne 1 ; then
                     enable_castor="no"
                 fi
                 ;;
