diff -ur root-6.18.04.orig/graf2d/gpadv7/v7/inc/ROOT/RColor.hxx root-6.18.04/graf2d/gpadv7/v7/inc/ROOT/RColor.hxx
--- root-6.18.04.orig/graf2d/gpadv7/v7/inc/ROOT/RColor.hxx	2019-09-11 15:36:49.000000000 +0200
+++ root-6.18.04/graf2d/gpadv7/v7/inc/ROOT/RColor.hxx	2020-01-23 11:27:58.637463837 +0100
@@ -17,6 +17,7 @@
 #define ROOT7_RColor
 
 #include <array>
+#include <string>
 #include <vector>
 
 namespace ROOT {
