diff -ur root-5.28.00.orig/config/Makefile.linuxppc64gcc root-5.28.00/config/Makefile.linuxppc64gcc
--- root-5.28.00.orig/config/Makefile.linuxppc64gcc	2010-12-14 15:20:26.000000000 +0100
+++ root-5.28.00/config/Makefile.linuxppc64gcc	2011-01-14 20:53:19.298991959 +0100
@@ -38,7 +38,7 @@
 # System libraries:
 SYSLIBS       = -lm -ldl $(OSTHREADLIBDIR) $(OSTHREADLIB) \
                 $(ALLOCLIBDIR) $(ALLOCLIB)
-XLIBS         = $(XPMLIBDIR) $(XPMLIB) $(X11LIBDIR) -lXext -lX11 -lSM -lICE
+XLIBS         = $(XPMLIBDIR) $(XPMLIB) $(X11LIBDIR) -lXext -lX11
 CILIBS        = -lm -ldl
 CRYPTLIBS     = -lcrypt
 
diff -ur root-5.28.00.orig/config/Makefile.linuxppcgcc root-5.28.00/config/Makefile.linuxppcgcc
--- root-5.28.00.orig/config/Makefile.linuxppcgcc	2010-12-14 15:20:26.000000000 +0100
+++ root-5.28.00/config/Makefile.linuxppcgcc	2011-01-14 20:52:57.826288692 +0100
@@ -38,7 +38,7 @@
 # System libraries:
 SYSLIBS       = -lm -ldl $(OSTHREADLIBDIR) $(OSTHREADLIB) \
                 $(ALLOCLIBDIR) $(ALLOCLIB)
-XLIBS         = $(XPMLIBDIR) $(XPMLIB) $(X11LIBDIR) -lXext -lX11 -lSM -lICE
+XLIBS         = $(XPMLIBDIR) $(XPMLIB) $(X11LIBDIR) -lXext -lX11
 CILIBS        = -lm -ldl
 CRYPTLIBS     = -lcrypt
 
