diff --git a/tutorials/mlp/mlpHiggs.C b/tutorials/mlp/mlpHiggs.C
index 8c5743f..49d3ca2 100644
--- a/tutorials/mlp/mlpHiggs.C
+++ b/tutorials/mlp/mlpHiggs.C
@@ -18,6 +18,8 @@ void mlpHiggs(Int_t ntrain=100) {
    TFile *input = 0;
    if (!gSystem->AccessPathName(fname)) {
       input = TFile::Open(fname);
+   } else if (!gSystem->AccessPathName(Form("%s/mlp/%s", TROOT::GetTutorialDir().Data(), fname))) {
+      input = TFile::Open(Form("%s/mlp/%s", TROOT::GetTutorialDir().Data(), fname));
    } else {
       printf("accessing %s file from http://root.cern.ch/files\n",fname);
       input = TFile::Open(Form("http://root.cern.ch/files/%s",fname));
diff --git a/tutorials/quadp/portfolio.C b/tutorials/quadp/portfolio.C
index 9f8bb8a..6c39446 100644
--- a/tutorials/quadp/portfolio.C
+++ b/tutorials/quadp/portfolio.C
@@ -291,6 +291,8 @@ void portfolio()
    TFile *f = 0;
    if (!gSystem->AccessPathName(fname)) {
       f = TFile::Open(fname);
+   } else if (!gSystem->AccessPathName(Form("%s/quadp/%s", TROOT::GetTutorialDir().Data(), fname))) {
+      f = TFile::Open(Form("%s/quadp/%s", TROOT::GetTutorialDir().Data(), fname));
    } else {
       printf("accessing %s file from http://root.cern.ch/files\n",fname);
       f = TFile::Open(Form("http://root.cern.ch/files/%s",fname));
@@ -314,7 +316,7 @@ void portfolio()
       for (Int_t j = 0; j <= i; j++) {
          Double_t sum = 0.;
          for (Int_t k = 0; k < nrData; k++)
-         sum += (data[i][k]-r[i])*(data[j][k]-r[j]);
+            sum += (data[i][k]-r[i])*(data[j][k]-r[j]);
          Covar(i,j) = Covar(j,i) = sum/nrData;
       }
    }
