diff --git a/core/base/CMakeLists.txt b/core/base/CMakeLists.txt
index b80f366..35e0f70 100644
--- a/core/base/CMakeLists.txt
+++ b/core/base/CMakeLists.txt
@@ -2,7 +2,10 @@
 # CMakeLists.txt file for building ROOT core/base package
 ############################################################################
 
-ROOT_GLOB_HEADERS(Base_dict_headers ${CMAKE_CURRENT_SOURCE_DIR}/inc/T*.h)
+ROOT_GLOB_HEADERS(Base_dict_headers ${CMAKE_CURRENT_SOURCE_DIR}/inc/T*.h
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/inc/GuiTypes.h 
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/inc/KeySymbols.h 
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/inc/Buttons.h)
 
 if(root7)
     set(root7src v7/src/*.cxx)
