diff -ur root-5.34.05.orig/configure root-5.34.05/configure
--- root-5.34.05.orig/configure	2013-02-14 23:27:56.000000000 +0100
+++ root-5.34.05/configure	2013-03-07 19:40:36.108220359 +0100
@@ -3639,7 +3639,7 @@
     gvizlib1=$found_lib
     gvizlib1raw=$found_raw_lib
     gvizlibir=$found_dir
-    check_library "libgraph graph" "no" "$gvizlibdir" \
+    check_library "libgraph graph libcgraph cgraph" "no" "$gvizlibdir" \
         $GVIZ ${GVIZ:+$GVIZ/lib/graphviz} ${GVIZ:+$GVIZ/lib} ${GVIZ:+$GVIZ/.libs} \
         ${finkdir:+$finkdir/lib} \
         /usr/local/lib /usr/lib
diff -ur root-5.34.05.orig/graf2d/gviz/inc/TGraphEdge.h root-5.34.05/graf2d/gviz/inc/TGraphEdge.h
--- root-5.34.05.orig/graf2d/gviz/inc/TGraphEdge.h	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/inc/TGraphEdge.h	2013-03-07 19:35:10.936220359 +0100
@@ -20,8 +20,13 @@
 #include "TAttLine.h"
 #endif
 
+#if !defined(__CINT__)
+#include <types.h>
+#else
 struct Agraph_t;
 struct Agedge_t;
+#endif
+
 class  TGraphNode;
 
 //////////////////////////////////////////////////////////////////////////
diff -ur root-5.34.05.orig/graf2d/gviz/inc/TGraphNode.h root-5.34.05/graf2d/gviz/inc/TGraphNode.h
--- root-5.34.05.orig/graf2d/gviz/inc/TGraphNode.h	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/inc/TGraphNode.h	2013-03-07 19:35:10.946220359 +0100
@@ -28,9 +28,12 @@
 #include "TAttLine.h"
 #endif
 
+#if !defined(__CINT__)
+#include <types.h>
+#else
 struct Agraph_t;
 struct Agnode_t;
-
+#endif
 
 //////////////////////////////////////////////////////////////////////////
 //                                                                      //
diff -ur root-5.34.05.orig/graf2d/gviz/inc/TGraphStruct.h root-5.34.05/graf2d/gviz/inc/TGraphStruct.h
--- root-5.34.05.orig/graf2d/gviz/inc/TGraphStruct.h	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/inc/TGraphStruct.h	2013-03-07 19:35:10.954220359 +0100
@@ -25,7 +25,12 @@
 #include "TList.h"
 #endif
 
+#if !defined(__CINT__)
+#include <types.h>
+#else
 struct Agraph_t;
+#endif
+
 struct GVC_s;
 
 //////////////////////////////////////////////////////////////////////////
diff -ur root-5.34.05.orig/graf2d/gviz/Module.mk root-5.34.05/graf2d/gviz/Module.mk
--- root-5.34.05.orig/graf2d/gviz/Module.mk	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/Module.mk	2013-03-07 19:35:10.921220359 +0100
@@ -68,4 +68,4 @@
 distclean::     distclean-$(MODNAME)
 
 ##### extra rules ######
-$(GVIZO): CXXFLAGS += $(GRAPHVIZINCDIR:%=-I%) $(GRAPHVIZCFLAGS)
+$(GVIZO) $(GVIZDO): CXXFLAGS += $(GRAPHVIZINCDIR:%=-I%) $(GRAPHVIZCFLAGS)
diff -ur root-5.34.05.orig/graf2d/gviz/src/TGraphEdge.cxx root-5.34.05/graf2d/gviz/src/TGraphEdge.cxx
--- root-5.34.05.orig/graf2d/gviz/src/TGraphEdge.cxx	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/src/TGraphEdge.cxx	2013-03-07 19:35:10.892220359 +0100
@@ -81,7 +81,11 @@
    if (gv) {
       Agnode_t *n1 = fNode1->GetGVNode();
       Agnode_t *n2 = fNode2->GetGVNode();
+#ifdef WITH_CGRAPH
+      fGVEdge = agedge(gv, n1, n2, NULL, 1);
+#else
       fGVEdge = agedge(gv, n1, n2);
+#endif
    } else {
       Error("CreateGVEdge","Invalid graphviz graph");
    }
diff -ur root-5.34.05.orig/graf2d/gviz/src/TGraphNode.cxx root-5.34.05/graf2d/gviz/src/TGraphNode.cxx
--- root-5.34.05.orig/graf2d/gviz/src/TGraphNode.cxx	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/src/TGraphNode.cxx	2013-03-07 19:35:10.904220359 +0100
@@ -65,7 +65,11 @@
    // Create the GraphViz node into the GraphViz data structure gv.
 
    if (gv) {
+#ifdef WITH_CGRAPH
+      fGVNode = agnode(gv, (char *)GetName(), 1);
+#else
       fGVNode = agnode(gv, (char *)GetName());
+#endif
    } else {
       Error("CreateGVNode","Invalid graphviz graph");
    }
diff -ur root-5.34.05.orig/graf2d/gviz/src/TGraphStruct.cxx root-5.34.05/graf2d/gviz/src/TGraphStruct.cxx
--- root-5.34.05.orig/graf2d/gviz/src/TGraphStruct.cxx	2013-02-14 23:27:53.000000000 +0100
+++ root-5.34.05/graf2d/gviz/src/TGraphStruct.cxx	2013-03-07 19:35:10.913220359 +0100
@@ -226,7 +226,11 @@
       gvFreeLayout(fGVC,fGVGraph);
       agclose(fGVGraph);
    }
+#ifdef WITH_CGRAPH
+   fGVGraph = agopen((char*)"GVGraph", Agdirected, 0);
+#else
    fGVGraph = agopen((char*)"GVGraph", AGDIGRAPH);
+#endif
 
    // Put the GV nodes into the GV graph
    if (fNodes) {
