diff -ur root-6.06.02.orig/documentation/users-guide/PhysicsVectors.md root-6.06.02/documentation/users-guide/PhysicsVectors.md
--- root-6.06.02.orig/documentation/users-guide/PhysicsVectors.md	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/documentation/users-guide/PhysicsVectors.md	2016-03-07 19:15:32.945004806 +0100
@@ -5,7 +5,7 @@
 and their rotation algorithms. The classes were ported to root from
 CLHEP see:
 
-<http://wwwinfo.cern.ch/asd/lhc++/clhep/manual/UserGuide/Vector/vector.html>
+<http://www.cern.ch/clhep/manual/UserGuide/Vector/vector.html>
 
 ## The Physics Vector Classes
 
diff -ur root-6.06.02.orig/hist/hist/src/TVirtualFitter.cxx root-6.06.02/hist/hist/src/TVirtualFitter.cxx
--- root-6.06.02.orig/hist/hist/src/TVirtualFitter.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/hist/hist/src/TVirtualFitter.cxx	2016-03-07 19:15:32.945004806 +0100
@@ -331,7 +331,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// static: Set the maximum number of function calls for the minimization algorithm
 /// For example for MIGRAD this is the maxcalls value passed as first argument
-/// (see http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/node18.html )
+/// (see https://cern-tex.web.cern.ch/cern-tex/minuit/node18.html )
 
 void TVirtualFitter::SetMaxIterations(Int_t niter)
 {
@@ -341,7 +341,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// static: Set the Error Definition (default=1)
 /// For Minuit this is the value passed with the "SET ERR" command
-/// (see http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/node18.html)
+/// (see https://cern-tex.web.cern.ch/cern-tex/minuit/node18.html)
 
 void TVirtualFitter::SetErrorDef(Double_t errdef)
 {
@@ -356,7 +356,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 /// static: Set the tolerance used in the minimization algorithm
 /// For example for MIGRAD this is tolerance value passed as second argument
-/// (see http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/node18.html )
+/// (see https://cern-tex.web.cern.ch/cern-tex/minuit/node18.html )
 
 void TVirtualFitter::SetPrecision(Double_t prec)
 {
diff -ur root-6.06.02.orig/man/man1/g2root.1 root-6.06.02/man/man1/g2root.1
--- root-6.06.02.orig/man/man1/g2root.1	2016-03-09 23:32:55.884697035 +0100
+++ root-6.06.02/man/man1/g2root.1	2016-03-07 19:15:34.837027205 +0100
@@ -13,7 +13,7 @@
 You can convert a
 .B GEANT
 (see
-.I http://wwwinfo.cern.ch/asd/geant/index.html
+.I http://geant.cern.ch/
 fore more on
 .BR GEANT )
 geometry to
diff -ur root-6.06.02.orig/man/man1/g2rootold.1 root-6.06.02/man/man1/g2rootold.1
--- root-6.06.02.orig/man/man1/g2rootold.1	2016-03-09 23:32:55.886697054 +0100
+++ root-6.06.02/man/man1/g2rootold.1	2016-03-07 19:15:34.838027217 +0100
@@ -5,15 +5,15 @@
 .\" NAME should be all caps, SECTION should be 1-8, maybe w/ subsection
 .\" other parms are allowed: see man(7), man(1)
 .SH NAME
-g2rootold \- convert GEANT geomtry files to ROOT files
+g2rootold \- convert GEANT geometry files to ROOT files
 .SH SYNOPSIS
-.B g2rootoldd
+.B g2rootold
 .I [-f map_name] geant_name macro_name
 .SH "DESCRIPTION"
 You can convert a
 .B GEANT
 (see
-.I http://wwwinfo.cern.ch/asd/geant/index.html
+.I http://geant.cern.ch/
 fore more on
 .BR GEANT )
 geometry to
@@ -103,9 +103,9 @@
 .I JDIGI.
 .SH "SEE ALSO"
 .RS
-.IR root (1),
-.IR g2root (1),
-.IR h2root(1),
+.IR root(1) ,
+.IR g2root(1) ,
+.IR h2root(1) ,
 .RE
 and
 .RS
diff -ur root-6.06.02.orig/math/doc/v520/index.html root-6.06.02/math/doc/v520/index.html
--- root-6.06.02.orig/math/doc/v520/index.html	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/doc/v520/index.html	2016-03-07 19:15:34.407022114 +0100
@@ -74,7 +74,7 @@
 
     <ul>
   <li><tt>GaussIntegrator</tt> and <tt>GaussLegendreIntegrator</tt> for numerical integration of one-dimensional functions. The first class uses Gaussian  8 and 16 point  quadrature approximation, it provides the translation of the CERNLIB algorithm
-  <a href="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/d103/top.html">DGAUSS</a> by Sigfried Kolbig,  and it is used by the <tt>TF1::Integral</tt> method. The second one uses the Gauss Legendre quadrature formula. It is used by the <tt>TF1::IntegralFast</tt> method.
+  <a href="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/d103/top.html">DGAUSS</a> by Sigfried Kolbig,  and it is used by the <tt>TF1::Integral</tt> method. The second one uses the Gauss Legendre quadrature formula. It is used by the <tt>TF1::IntegralFast</tt> method.
   These classes implement both the same  virtual interface as the adaptive integration methods provided by the MathMore library.  They can all be created and used easily via the common class <tt>ROOT::Math::IntegratorOneDim</tt> providing the interfaces for numerical integration.
 New template methods have been also included in the common Integration class in order to be able to integrate automatically any C++ callable object.
 </li>
diff -ur root-6.06.02.orig/math/mathcore/inc/Math/PdfFuncMathCore.h root-6.06.02/math/mathcore/inc/Math/PdfFuncMathCore.h
--- root-6.06.02.orig/math/mathcore/inc/Math/PdfFuncMathCore.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathcore/inc/Math/PdfFuncMathCore.h	2016-03-07 19:15:34.407022114 +0100
@@ -301,7 +301,7 @@
    <A HREF="http://dx.doi.org/10.1016/0010-4655(84)90085-7">Computer Phys. Comm. 31 (1984) 97-111</A>
    <A HREF="http://dx.doi.org/10.1016/j.cpc.2008.03.002">[Erratum-ibid. 178 (2008) 972]</A>.
    The same algorithms as in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g110/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g110/top.html">
    CERNLIB</A> (DENLAN)  is used
 
    @param x The argument \f$x\f$
diff -ur root-6.06.02.orig/math/mathcore/inc/Math/ProbFuncMathCore.h root-6.06.02/math/mathcore/inc/Math/ProbFuncMathCore.h
--- root-6.06.02.orig/math/mathcore/inc/Math/ProbFuncMathCore.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathcore/inc/Math/ProbFuncMathCore.h	2016-03-07 19:15:34.407022114 +0100
@@ -381,7 +381,7 @@
    <A HREF="http://dx.doi.org/10.1016/0010-4655(84)90085-7">Computer Phys. Comm. 31 (1984) 97-111</A>
    <A HREF="http://dx.doi.org/10.1016/j.cpc.2008.03.002">[Erratum-ibid. 178 (2008) 972]</A>.
    The same algorithms as in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g110/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g110/top.html">
    CERNLIB</A> (DISLAN) is used.
 
    @param x The argument \f$x\f$
@@ -756,7 +756,7 @@
    <A HREF="http://dx.doi.org/10.1016/0010-4655(84)90085-7">Computer Phys. Comm. 31 (1984) 97-111</A>
    <A HREF="http://dx.doi.org/10.1016/j.cpc.2008.03.002">[Erratum-ibid. 178 (2008) 972]</A>.
    The same algorithms as in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g110/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g110/top.html">
    CERNLIB</A> (XM1LAN)  is used
 
    @param x The argument \f$x\f$
@@ -783,7 +783,7 @@
    <A HREF="http://dx.doi.org/10.1016/0010-4655(84)90085-7">Computer Phys. Comm. 31 (1984) 97-111</A>
    <A HREF="http://dx.doi.org/10.1016/j.cpc.2008.03.002">[Erratum-ibid. 178 (2008) 972]</A>.
    The same algorithms as in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g110/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g110/top.html">
    CERNLIB</A> (XM1LAN)  is used
 
    @param x The argument \f$x\f$
diff -ur root-6.06.02.orig/math/mathcore/inc/Math/QuantFuncMathCore.h root-6.06.02/math/mathcore/inc/Math/QuantFuncMathCore.h
--- root-6.06.02.orig/math/mathcore/inc/Math/QuantFuncMathCore.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathcore/inc/Math/QuantFuncMathCore.h	2016-03-07 19:15:34.407022114 +0100
@@ -528,7 +528,7 @@
    <A HREF="http://dx.doi.org/10.1016/0010-4655(84)90085-7">Computer Phys. Comm. 31 (1984) 97-111</A>
    <A HREF="http://dx.doi.org/10.1016/j.cpc.2008.03.002">[Erratum-ibid. 178 (2008) 972]</A>.
    The same algorithms as in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g110/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g110/top.html">
    CERNLIB</A> (RANLAN) is used.
 
    @param z The argument \f$z\f$
diff -ur root-6.06.02.orig/math/mathcore/inc/Math/SpecFuncMathCore.h root-6.06.02/math/mathcore/inc/Math/SpecFuncMathCore.h
--- root-6.06.02.orig/math/mathcore/inc/Math/SpecFuncMathCore.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathcore/inc/Math/SpecFuncMathCore.h	2016-03-07 19:15:34.408022126 +0100
@@ -193,7 +193,7 @@
   For detailed description see
   <A HREF="http://mathworld.wolfram.com/SineIntegral.html">
   Mathworld</A>. The implementation used is that of
-  <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c336/top.html">
+  <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c336/top.html">
   CERNLIB</A>,
   based on Y.L. Luke, The special functions and their approximations, v.II, (Academic Press, New York l969) 325-326.
 
@@ -219,7 +219,7 @@
   For detailed description see
   <A HREF="http://mathworld.wolfram.com/CosineIntegral.html">
   Mathworld</A>. The implementation used is that of
-  <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c336/top.html">
+  <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c336/top.html">
   CERNLIB</A>,
   based on Y.L. Luke, The special functions and their approximations, v.II, (Academic Press, New York l969) 325-326.
 
diff -ur root-6.06.02.orig/math/mathmore/inc/Math/SpecFuncMathMore.h root-6.06.02/math/mathmore/inc/Math/SpecFuncMathMore.h
--- root-6.06.02.orig/math/mathmore/inc/Math/SpecFuncMathMore.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathmore/inc/Math/SpecFuncMathMore.h	2016-03-07 19:15:34.681025358 +0100
@@ -193,7 +193,7 @@
   (<A HREF="http://www.gnu.org/software/gsl/manual/gsl-ref_7.html#SEC95">
   GSL</A>, <A HREF="http://planetmath.org/encyclopedia/EllipticIntegralsAndJacobiEllipticFunctions.html">
   Planetmath</A> and
-  <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c346/top.html">
+  <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c346/top.html">
   CERNLIB</A>) use the + sign in front of n in the denominator. In
   order to be C++ compliant, the present library uses the former
   convention. The implementation used is that of
@@ -408,7 +408,7 @@
   (<A HREF="http://www.gnu.org/software/gsl/manual/gsl-ref_7.html#SEC95">
   GSL</A>, <A HREF="http://planetmath.org/encyclopedia/EllipticIntegralsAndJacobiEllipticFunctions.html">
   Planetmath</A> and
-  <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c346/top.html">
+  <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c346/top.html">
   CERNLIB</A>) use the + sign in front of n in the denominator. In
   order to be C++ compliant, the present library uses the former
   convention. The implementation used is that of
diff -ur root-6.06.02.orig/math/mathmore/inc/Math/VavilovAccurate.h root-6.06.02/math/mathmore/inc/Math/VavilovAccurate.h
--- root-6.06.02.orig/math/mathmore/inc/Math/VavilovAccurate.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathmore/inc/Math/VavilovAccurate.h	2016-03-07 19:15:34.681025358 +0100
@@ -81,7 +81,7 @@
    B. Schorr, Programs for the Landau and the Vavilov distributions and the corresponding random numbers,
    <A HREF="http://dx.doi.org/10.1016/0010-4655(74)90091-5">Computer Phys. Comm. 7 (1974) 215-224</A>,
    which has been implemented in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g116/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g116/top.html">
    CERNLIB (G116)</A>.
 
    The class stores coefficients needed to calculate \f$p(\lambda; \kappa, \beta^2)\f$
diff -ur root-6.06.02.orig/math/mathmore/inc/Math/VavilovFast.h root-6.06.02/math/mathmore/inc/Math/VavilovFast.h
--- root-6.06.02.orig/math/mathmore/inc/Math/VavilovFast.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathmore/inc/Math/VavilovFast.h	2016-03-07 19:15:34.681025358 +0100
@@ -90,7 +90,7 @@
    A. Rotondi and P. Montagna, Fast calculation of Vavilov distribution,
    <A HREF="http://dx.doi.org/10.1016/0168-583X(90)90749-K">Nucl. Instr. and Meth. B47 (1990) 215-224</A>,
    which has been implemented in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g115/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g115/top.html">
    CERNLIB (G115)</A>.
 
    The class stores coefficients needed to calculate \f$p(\lambda; \kappa, \beta^2)\f$
diff -ur root-6.06.02.orig/math/mathmore/inc/Math/Vavilov.h root-6.06.02/math/mathmore/inc/Math/Vavilov.h
--- root-6.06.02.orig/math/mathmore/inc/Math/Vavilov.h	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathmore/inc/Math/Vavilov.h	2016-03-07 19:15:34.681025358 +0100
@@ -92,14 +92,14 @@
    A. Rotondi and P. Montagna, Fast calculation of Vavilov distribution,
    <A HREF="http://dx.doi.org/10.1016/0168-583X(90)90749-K">Nucl. Instr. and Meth. B47 (1990) 215-224</A>,
    which has been implemented in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g115/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g115/top.html">
    CERNLIB (G115)</A>.
 
    - VavilovAccurate uses the algorithm by
    B. Schorr, Programs for the Landau and the Vavilov distributions and the corresponding random numbers,
    <A HREF="http://dx.doi.org/10.1016/0010-4655(74)90091-5">Computer Phys. Comm. 7 (1974) 215-224</A>,
    which has been implemented in
-   <A HREF="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/g116/top.html">
+   <A HREF="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/g116/top.html">
    CERNLIB (G116)</A>.
 
    Both subclasses store coefficients needed to calculate \f$p(\lambda; \kappa, \beta^2)\f$
diff -ur root-6.06.02.orig/math/mathmore/src/SpecFuncMathMore.cxx root-6.06.02/math/mathmore/src/SpecFuncMathMore.cxx
--- root-6.06.02.orig/math/mathmore/src/SpecFuncMathMore.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/mathmore/src/SpecFuncMathMore.cxx	2016-03-07 19:15:34.681025358 +0100
@@ -103,7 +103,7 @@
      http://planetmath.org/encyclopedia/EllipticIntegralsAndJacobiEllipticFunctions.html
 
 - CERNLIB
-     http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c346/top.html
+     https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c346/top.html
 
    while the latter is used by
 
@@ -239,7 +239,7 @@
      http://planetmath.org/encyclopedia/EllipticIntegralsAndJacobiEllipticFunctions.html
 
 - CERNLIB
-     http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/c346/top.html
+     https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/c346/top.html
 
    while the latter is used by
 
diff -ur root-6.06.02.orig/math/minuit/doc/index.txt root-6.06.02/math/minuit/doc/index.txt
--- root-6.06.02.orig/math/minuit/doc/index.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/minuit/doc/index.txt	2016-03-07 19:15:34.681025358 +0100
@@ -5,7 +5,7 @@
 <li>The main TMinuit class is documented in class TMinuit.</li>
 <li><a href="ftp://root.cern.ch/root/doc/5FittingHistograms.pdf">The Chapter about Fitting Histogram in the Users Guide
     </a></li>
-<li><a href="http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/minmain.html">The MINUIT documentation in CERNLIB
+<li><a href="https://cern-tex.web.cern.ch/cern-tex/minuit/minmain.html">The MINUIT documentation in CERNLIB
     </a></li>
 <li><a href="http://root.cern.ch/drupal/content/howtos#fit">How to Fit Histograms
       </A></li>
diff -ur root-6.06.02.orig/math/minuit2/doc/index.txt root-6.06.02/math/minuit2/doc/index.txt
--- root-6.06.02.orig/math/minuit2/doc/index.txt	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/minuit2/doc/index.txt	2016-03-07 19:15:34.682025370 +0100
@@ -13,7 +13,7 @@
 <p>
 <ol>
 <li>
-F. James, <em>Fortran MINUIT Reference Manual</em> (<a href="http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/minmain.html">html</a>); </li>
+F. James, <em>Fortran MINUIT Reference Manual</em> (<a href="https://cern-tex.web.cern.ch/cern-tex/minuit/minmain.html">html</a>); </li>
 <li>
 F. James and M. Winkler, <em>C++ MINUIT User's Guide</em> (<a href="http://seal.cern.ch/documents/minuit/mnusersguide.pdf">pdf</a>); </li>
 <li>
diff -ur root-6.06.02.orig/math/minuit2/doc/Minuit2.html root-6.06.02/math/minuit2/doc/Minuit2.html
--- root-6.06.02.orig/math/minuit2/doc/Minuit2.html	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/minuit2/doc/Minuit2.html	2016-03-07 19:15:34.681025358 +0100
@@ -32,7 +32,7 @@
 
 <h2>References</h2>
 <ol>
-<li>F. James, <em>Fortran MINUIT Reference Manual</em> (<a href="http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/minmain.html">html</a>);
+<li>F. James, <em>Fortran MINUIT Reference Manual</em> (<a href="https://cern-tex.web.cern.ch/cern-tex/minuit/minmain.html">html</a>);
 <li>F. James and M. Winkler, <em>C++ MINUIT User's Guide</em> (<a href="https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.html">html</a> and
   <a href="https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.pdf">pdf</a>);
 <li>F. James, <em>Minuit Tutorial on Function Minimization</em> (<a href="http://seal.cern.ch/documents/minuit/mntutorial.pdf">pdf</a>);
diff -ur root-6.06.02.orig/math/minuit2/doc/Minuit2.md root-6.06.02/math/minuit2/doc/Minuit2.md
--- root-6.06.02.orig/math/minuit2/doc/Minuit2.md	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/minuit2/doc/Minuit2.md	2016-03-07 19:15:34.682025370 +0100
@@ -14,7 +14,7 @@
 
 ## References
 
-1.  F. James, _Fortran MINUIT Reference Manual_ ([html](http://wwwasdoc.web.cern.ch/wwwasdoc/minuit/minmain.html));
+1.  F. James, _Fortran MINUIT Reference Manual_ ([html](https://cern-tex.web.cern.ch/cern-tex/minuit/minmain.html));
 2.  F. James and M. Winkler, _C++ MINUIT User's Guide_ ([html](https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.html) and [pdf](https://root.cern.ch/root/htmldoc/guides/minuit2/Minuit2.pdf));
 3.  F. James, _Minuit Tutorial on Function Minimization_ ([pdf](http://seal.cern.ch/documents/minuit/mntutorial.pdf));
 4.  F. James, _The Interpretation of Errors in Minuit_ ([pdf](http://seal.cern.ch/documents/minuit/mnerror.pdf));
diff -ur root-6.06.02.orig/math/smatrix/doc/SMatrixClass.html root-6.06.02/math/smatrix/doc/SMatrixClass.html
--- root-6.06.02.orig/math/smatrix/doc/SMatrixClass.html	2016-03-11 18:29:45.555122263 +0100
+++ root-6.06.02/math/smatrix/doc/SMatrixClass.html	2016-03-07 19:15:34.682025370 +0100
@@ -163,7 +163,7 @@
 <h3>Linear Algebra Functions</h3>
 
 Only limited linear algebra functionality is available for SMatrix. It is possible for squared matrices NxN,  to find the inverse or to calculate the determinant. Different inversion algorithms are used if the matrix is smaller than 6x6 or if it is symmetric.
-In the case of a small matrix, a faster direct inversion is used. For a large (N > 6)  symmetric matrix the Bunch-Kaufman diagonal pivoting method is used while for a large (N > 6) general matrix an LU factorization is performed using the same algorithm as in the CERNLIB routine <a href="http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f010/top.html">dinv</a>.
+In the case of a small matrix, a faster direct inversion is used. For a large (N > 6)  symmetric matrix the Bunch-Kaufman diagonal pivoting method is used while for a large (N > 6) general matrix an LU factorization is performed using the same algorithm as in the CERNLIB routine <a href="https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f010/top.html">dinv</a>.
 <pre>
 <em>//  Invert a NxN matrix. The inverted matrix replace the existing one and returns if the result is successful</em>
 bool ret = m.<strong>Invert</strong>()
diff -ur root-6.06.02.orig/math/smatrix/doc/SMatrixClass.md root-6.06.02/math/smatrix/doc/SMatrixClass.md
--- root-6.06.02.orig/math/smatrix/doc/SMatrixClass.md	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/math/smatrix/doc/SMatrixClass.md	2016-03-07 19:15:34.682025370 +0100
@@ -132,7 +132,7 @@
 
 ### Linear Algebra Functions
 
-Only limited linear algebra functionality is available for SMatrix. It is possible for squared matrices NxN, to find the inverse or to calculate the determinant. Different inversion algorithms are used if the matrix is smaller than 6x6 or if it is symmetric. In the case of a small matrix, a faster direct inversion is used. For a large (N > 6) symmetric matrix the Bunch-Kaufman diagonal pivoting method is used while for a large (N > 6) general matrix an LU factorization is performed using the same algorithm as in the CERNLIB routine [dinv](http://wwwasdoc.web.cern.ch/wwwasdoc/shortwrupsdir/f010/top.html).
+Only limited linear algebra functionality is available for SMatrix. It is possible for squared matrices NxN, to find the inverse or to calculate the determinant. Different inversion algorithms are used if the matrix is smaller than 6x6 or if it is symmetric. In the case of a small matrix, a faster direct inversion is used. For a large (N > 6) symmetric matrix the Bunch-Kaufman diagonal pivoting method is used while for a large (N > 6) general matrix an LU factorization is performed using the same algorithm as in the CERNLIB routine [dinv](https://cern-tex.web.cern.ch/cern-tex/shortwrupsdir/f010/top.html).
 
 <pre>_//  Invert a NxN matrix. The inverted matrix replace the existing one and returns if the result is successful_
 bool ret = m.**Invert**()  
diff -ur root-6.06.02.orig/misc/table/src/TPolyLineShape.cxx root-6.06.02/misc/table/src/TPolyLineShape.cxx
--- root-6.06.02.orig/misc/table/src/TPolyLineShape.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/misc/table/src/TPolyLineShape.cxx	2016-03-10 22:03:03.211002438 +0100
@@ -47,8 +47,8 @@
 //           V               v      v                     v      v                  //
 //      TPolyLineShape     TBRIK  TTUBE                 TPCON  TTRD1                //
 //       |        ^                                                                 //
-//       |        |       begin_html <a href="http://root.cern.ch/root/html/TShape.html#TShape:description">R  O  O  T        S  H  A  P  E  S</a>end_html                          //
-//       V        |       (see begin_html <a href="http://wwwinfo.cern.ch/asdoc/geant_html3/node109.html#SECTION041000000000000000000000">GEANT 3.21 shapes</a>end_html as well)                           //
+//       |        |       <a href="http://root.cern.ch/root/html/TShape.html#TShape:description">R  O  O  T        S  H  A  P  E  S</a>                          //
+//       V        |       (see <a href="https://cern-tex.web.cern.ch/cern-tex/geant_html3/node109.html#SECTION041000000000000000000000">GEANT 3.21 shapes</a> as well)                           //
 // St_PolyLine3D  |                                                                 //
 //                |                                                                 //
 //           TPoints3DABC                                                           //
diff -ur root-6.06.02.orig/misc/table/src/TVolume.cxx root-6.06.02/misc/table/src/TVolume.cxx
--- root-6.06.02.orig/misc/table/src/TVolume.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/misc/table/src/TVolume.cxx	2016-03-09 06:11:44.976464126 +0100
@@ -142,7 +142,7 @@
 ///                            10 - this unvisible, but sons are visible
 ///                            01 - this visible but sons
 ///                            11 - neither this nor its sons are visible
-/// Maps the value of the visibility flag to begin_html <a href="http://wwwinfo.cern.ch/asdoc/geant_html3/node128.html#SECTION056000000000000000000000">GEANT 3.21 "volume attributes"</a>end_html
+/// Maps the value of the visibility flag to <a href="https://cern-tex.web.cern.ch/cern-tex/geant_html3/node128.html#SECTION056000000000000000000000">GEANT 3.21 "volume attributes"</a>
 
 Int_t TVolume::MapStNode2GEANTVis(ENodeSEEN  vis)
 {
@@ -150,11 +150,12 @@
    return mapVis[vis];
 }
 
-//______________________________________________________________________________
-//ENodeSEEN TVolume::MapGEANT2StNodeVis(Int_t vis)
+////////////////////////////////////////////////////////////////////////////////
+/// ENodeSEEN TVolume::MapGEANT2StNodeVis(Int_t vis)
+/// Maps the value of <a href="https://cern-tex.web.cern.ch/cern-tex/geant_html3/node128.html#SECTION056000000000000000000000">GEANT 3.21 "volume attributes"</a> to the visibility flag
+
 Int_t TVolume::MapGEANT2StNodeVis(Int_t vis)
 {
-// Maps the value of begin_html <a href="http://wwwinfo.cern.ch/asdoc/geant_html3/node128.html#SECTION056000000000000000000000">GEANT 3.21 "volume attributes"</a>end_html to the visibility flag
    const Int_t mapVis[4] = {1, -2, 0, -1 };
    Int_t i;
 //  for (i =0; i<3;i++) if (mapVis[i] == vis) return (ENodeSEEN)i;
