From fb33e1aa0b7c0f2ea539a813df2d41b2c000f5b3 Mon Sep 17 00:00:00 2001
From: Mattias Ellert <mattias.ellert@physics.uu.se>
Date: Thu, 14 Feb 2019 13:09:02 +0100
Subject: [PATCH] Don't install eve7 files if eve7 is not built

---
 CMakeLists.txt | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74d24d4419..ce38150875 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -446,17 +446,16 @@ if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_INSTALL_PREFIX)
   else()
     install(DIRECTORY README DESTINATION ${CMAKE_INSTALL_DOCDIR})
   endif()
+  if (NOT root7)
+    set(excludeeve7 PATTERN "eve7" EXCLUDE)
+  endif()
   install(DIRECTORY etc/ DESTINATION ${CMAKE_INSTALL_SYSCONFDIR} USE_SOURCE_PERMISSIONS
                          PATTERN "system.rootrc" EXCLUDE
                          PATTERN "system.rootauthrc" EXCLUDE
                          PATTERN "system.rootdaemonrc" EXCLUDE
-                         PATTERN "rootd.rc.d" EXCLUDE
-                         PATTERN "proofd.rc.d" EXCLUDE
-                         PATTERN "rootd.xinetd" EXCLUDE
-                         PATTERN "proofd.xinetd" EXCLUDE
                          PATTERN "root.mimes" EXCLUDE
-                         PATTERN "cmake" EXCLUDE
-                         PATTERN "*.in" EXCLUDE)
+                         PATTERN "*.in" EXCLUDE
+                         ${excludeeve7})
   install(DIRECTORY fonts/  DESTINATION ${CMAKE_INSTALL_FONTDIR})
   install(DIRECTORY icons/  DESTINATION ${CMAKE_INSTALL_ICONDIR})
   install(DIRECTORY macros/ DESTINATION ${CMAKE_INSTALL_MACRODIR})
-- 
2.20.1

