diff -ur root-5.26.00d.orig/tutorials/tree/cernbuild.C root-5.26.00d/tutorials/tree/cernbuild.C
--- root-5.26.00d.orig/tutorials/tree/cernbuild.C	2010-08-24 13:29:47.000000000 +0200
+++ root-5.26.00d/tutorials/tree/cernbuild.C	2010-08-29 10:20:56.069801363 +0200
@@ -18,6 +18,7 @@
 
    //The input file cern.dat is a copy of the CERN staff data base
    //from 1988
+   TString filename = "cernstaff.root";
    TString dir = gSystem->UnixPathName(gInterpreter->GetCurrentMacroName());
    dir.ReplaceAll("cernbuild.C","");
    dir.ReplaceAll("/./","/");
@@ -70,7 +71,10 @@
    tree->Write();
 
    fclose(fp);
-   if (get) return hfile;
    delete hfile;
+   if (get) {
+     hfile = TFile::Open(filename);
+     return hfile;
+   }
    return 0;
 }
