diff -ur root-6.06.02.orig/core/base/src/TDirectory.cxx root-6.06.02/core/base/src/TDirectory.cxx
--- root-6.06.02.orig/core/base/src/TDirectory.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/base/src/TDirectory.cxx	2016-04-06 10:30:12.110817348 +0200
@@ -810,7 +810,7 @@
 ///      MyClass *obj = (MyClass*)directory->GetObjectChecked("some object of MyClass","MyClass"));
 /// ~~~
 ///  Note: We recommend using the method TDirectory::GetObject:
-/// ~~~ {.cpp}~
+/// ~~~ {.cpp}
 ///      MyClass *obj = 0;
 ///      directory->GetObject("some object inheriting from MyClass",obj);
 ///      if (obj) { ... we found what we are looking for ... }
diff -ur root-6.06.02.orig/core/base/src/TQObject.cxx root-6.06.02/core/base/src/TQObject.cxx
--- root-6.06.02.orig/core/base/src/TQObject.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/base/src/TQObject.cxx	2016-04-06 10:30:12.132817525 +0200
@@ -1040,7 +1040,7 @@
 ///
 ///       TQObject::Connect("TGButton", "Clicked()",
 ///                         "TH2F", myHist, "Draw(Option_t*)");
-///~~~
+/// ~~~
 /// cl == 0 - corresponds to function (interpereted or global)
 ///           the name of the function is defined by the slot string,
 ///           parameter receiver should be 0.
diff -ur root-6.06.02.orig/core/base/src/TSystem.cxx root-6.06.02/core/base/src/TSystem.cxx
--- root-6.06.02.orig/core/base/src/TSystem.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/base/src/TSystem.cxx	2016-04-06 10:30:12.133817533 +0200
@@ -2677,7 +2677,7 @@
 /// We allow them to type:
 /// ~~~ {.cpp}
 ///  .X myfunc.C++(arg1,arg2)
-///  ~~~
+/// ~~~
 /// or
 /// ~~~ {.cpp}
 ///  .X myfunc.C+(arg1,arg2)
@@ -3843,7 +3843,7 @@
 /// the use of ';' to separate several instructions. However, shell specific
 /// construct should be avoided. In particular this description can contain
 /// environment variables, like $ROOTSYS (or %ROOTSYS% on windows).
-///  ~~~ {.cpp}
+/// ~~~ {.cpp}
 /// Five special variables will be expanded before execution:
 ///   Variable name       Expands to
 ///   -------------       ----------
@@ -3915,7 +3915,7 @@
 /// IncludePath should contain the list of compiler flags to indicate where
 /// to find user defined header files. It is used to expand $IncludePath in
 /// the directives given to SetMakeSharedLib() and SetMakeExe(), e.g.:
-/// ~~~ {.cpp}
+///  ~~~ {.cpp}
 ///    gSystem->SetInclude("-I$ROOTSYS/include -Imydirectory/include");
 ///  ~~~
 /// the default value of IncludePath on Unix is:
diff -ur root-6.06.02.orig/core/cont/src/TClonesArray.cxx root-6.06.02/core/cont/src/TClonesArray.cxx
--- root-6.06.02.orig/core/cont/src/TClonesArray.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/cont/src/TClonesArray.cxx	2016-04-06 10:30:12.133817533 +0200
@@ -663,7 +663,7 @@
 /// Create an array of clone objects of class cl. The class must inherit from
 /// TObject. If the class defines an own operator delete(), make sure that
 /// it looks like this:
-///  ~~~ {.cpp}
+/// ~~~ {.cpp}
 ///    void MyClass::operator delete(void *vp)
 ///    {
 ///       if ((Long_t) vp != TObject::GetDtorOnly())
diff -ur root-6.06.02.orig/core/cont/src/TCollection.cxx root-6.06.02/core/cont/src/TCollection.cxx
--- root-6.06.02.orig/core/cont/src/TCollection.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/core/cont/src/TCollection.cxx	2016-04-06 10:30:12.134817541 +0200
@@ -321,7 +321,7 @@
 /// ~~~
 /// Otherwise override the `Print(Option_t *option, Int_t)`
 /// variant. Remember to declare:
-/// ~~~ {.cpp]
+/// ~~~ {.cpp}
 ///   using TCollection::Print;
 /// ~~~
 /// somewhere close to the method declaration.
diff -ur root-6.06.02.orig/documentation/users-guide/MathLibraries.md root-6.06.02/documentation/users-guide/MathLibraries.md
--- root-6.06.02.orig/documentation/users-guide/MathLibraries.md	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/documentation/users-guide/MathLibraries.md	2016-04-06 10:30:19.094873608 +0200
@@ -189,7 +189,7 @@
  int i;
  i = TMath::LocMin(10, &v[0]);
  std::cout << v[i] << std::endl;
- ```
+```
  
 Another example of these functions can be found in $ROOTSYS/tutorials/permute.C.
 
@@ -233,7 +233,7 @@
 
  // old-style
  mean = TMath::Mean(n, &v[0], &w[0]);
- ```
+```
  
 ### Special and Statistical Functions.
 
diff -ur root-6.06.02.orig/graf2d/asimage/src/TASImage.cxx root-6.06.02/graf2d/asimage/src/TASImage.cxx
--- root-6.06.02.orig/graf2d/asimage/src/TASImage.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf2d/asimage/src/TASImage.cxx	2016-04-06 10:30:12.135817549 +0200
@@ -5964,7 +5964,7 @@
 ///       im1->GetImageBuffer(&buf, &int, TImage::kXpm); /*raw buffer*/
 ///       TImage *im2 = TImage::Create();
 ///       im2->SetImageBuffer(&buf, TImage::kXpm);
-///  ~~~
+/// ~~~
 ///      2.  xpm as an array of strings (pre-parsed)
 /// ~~~ {.cpp}
 ///    For example:
diff -ur root-6.06.02.orig/graf2d/postscript/src/TPostScript.cxx root-6.06.02/graf2d/postscript/src/TPostScript.cxx
--- root-6.06.02.orig/graf2d/postscript/src/TPostScript.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf2d/postscript/src/TPostScript.cxx	2016-04-06 10:30:12.136817557 +0200
@@ -132,7 +132,7 @@
 because we clear only the pads and not the main canvas.
 Note that `c1->Update` must be called at the end of the first picture.
 
-~~~ {.cpp}
+ ~~~ {.cpp}
      {
         TFile *f1 = new TFile("hsimple.root");
         TCanvas *c1 = new TCanvas("c1");
diff -ur root-6.06.02.orig/graf3d/eve/src/TEveCalo3DGL.cxx root-6.06.02/graf3d/eve/src/TEveCalo3DGL.cxx
--- root-6.06.02.orig/graf3d/eve/src/TEveCalo3DGL.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf3d/eve/src/TEveCalo3DGL.cxx	2016-04-06 10:30:12.136817557 +0200
@@ -330,7 +330,7 @@
 ///    | 2-----|-3
 ///    |/      |/
 ///    1-------0
-///~~~
+/// ~~~
 
 void TEveCalo3DGL::RenderBox(const Float_t pnts[8]) const
 {
diff -ur root-6.06.02.orig/graf3d/g3d/src/TAxis3D.cxx root-6.06.02/graf3d/g3d/src/TAxis3D.cxx
--- root-6.06.02.orig/graf3d/g3d/src/TAxis3D.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/graf3d/g3d/src/TAxis3D.cxx	2016-04-06 10:30:12.136817557 +0200
@@ -71,7 +71,7 @@
 To change attributes of the rulers attached to the current Pad, one may
 query its pointer first:
 
-~~~ {.cpp]
+~~~ {.cpp}
 TAxis3D *axis = TAxis3D::GetPadAxis(); // Ask axis pointer
 if (axis) {
   TAxis3D::ToggleRulers()     // To pop axes down
diff -ur root-6.06.02.orig/hist/hist/src/TH2Poly.cxx root-6.06.02/hist/hist/src/TH2Poly.cxx
--- root-6.06.02.orig/hist/hist/src/TH2Poly.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/hist/hist/src/TH2Poly.cxx	2016-04-06 10:30:12.137817565 +0200
@@ -80,7 +80,7 @@
 many times.
 
 The following very simple macro shows how to build and fill a `TH2Poly`:
-~~~ {.cpp]
+~~~ {.cpp}
 {
     TH2Poly *h2p = new TH2Poly();
 
diff -ur root-6.06.02.orig/tree/tree/src/TTree.cxx root-6.06.02/tree/tree/src/TTree.cxx
--- root-6.06.02.orig/tree/tree/src/TTree.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/tree/tree/src/TTree.cxx	2016-04-06 10:30:12.138817573 +0200
@@ -2584,9 +2584,9 @@
 ///     T->Fill(); //loop
 ///     file->Write();
 ///     file->Close();
-///~~~
+/// ~~~
 /// but do the following:
-///~~~ {.cpp}
+/// ~~~ {.cpp}
 ///     TFile *file = new TFile("myfile.root","recreate");
 ///     TTree *T = new TTree("T","title");
 ///     T->Fill(); //loop
@@ -6487,7 +6487,7 @@
 /// ~~~ {.cpp}
 ///     TPrincipal *principal =
 ///     (TPrincipal*)gROOT->GetListOfSpecials()->FindObject("principal");
-///  ~~~
+/// ~~~
 
 TPrincipal* TTree::Principal(const char* varexp, const char* selection, Option_t* option, Long64_t nentries, Long64_t firstentry)
 {
diff -ur root-6.06.02.orig/tree/treeplayer/src/TTreePlayer.cxx root-6.06.02/tree/treeplayer/src/TTreePlayer.cxx
--- root-6.06.02.orig/tree/treeplayer/src/TTreePlayer.cxx	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/tree/treeplayer/src/TTreePlayer.cxx	2016-04-06 10:30:12.139817582 +0200
@@ -3004,6 +3004,7 @@
 /// The function return the status of the fit in the following form
 /// ~~~{.cpp}
 ///     fitResult = migradResult + 10*minosResult + 100*hesseResult + 1000*improveResult
+/// ~~~
 /// -  The fitResult is 0 is the fit is OK.
 /// -  The fitResult is negative in case of an error not connected with the fit.
 /// -  The number of entries used in the fit can be obtained via
