--- trunk/config/Makefile.depend	2010/07/27 12:56:50	34610
+++ trunk/config/Makefile.depend	2010/07/27 16:18:34	34625
@@ -39,6 +39,7 @@
 SESSIONVIEWERLIBDEPM   = $(PROOFLIB) $(GUILIB) $(HISTLIB) $(GPADLIB) $(GRAFLIB) \
                          $(TREELIB) $(MATHCORELIB)
 X3DLIBDEPM             = $(G3DLIB) $(GUILIB)
+CLARENSLIBDEPM         = $(MATHCORELIB)
 PEACLIBDEPM            = $(PROOFLIB) $(CLARENSLIB)
 PEACGUILIBDEPM         = $(GUILIB)
 EGLIBDEPM              = $(G3DLIB) $(GRAFLIB) $(GPADLIB) $(MATHCORELIB)
@@ -83,6 +84,7 @@
 CINTEXLIBDEPM          = $(REFLEXLIB)
 REFLEXDICTLIBDEPM      = $(REFLEXLIB)
 RAUTHLIBDEPM           = $(NETLIB) $(IOLIB)
+GLBSAUTHLIBDEPM        = $(RAUTHLIB)
 KRB5AUTHLIBDEPM        = $(RAUTHLIB) $(NETLIB)
 SRVAUTHLIBDEPM         = $(NETLIB)
 QTGSILIBDEPM           = $(GUILIB) $(GPADLIB)
@@ -149,6 +151,7 @@
 PROOFDRAWLIBDEP        = $(PROOFDRAWLIBDEPM)
 PROOFXLIBDEP           = $(PROOFXLIBDEPM)
 SESSIONVIEWERLIBDEP    = $(SESSIONVIEWERLIBDEPM)
+CLARENSLIBDEP          = $(CLARENSLIBDEPM)
 PEACLIBDEP             = $(PEACLIBDEPM)
 PEACGUILIBDEP          = $(PEACGUILIBDEPM)
 EGLIBDEP               = $(EGLIBDEPM)
@@ -180,6 +183,7 @@
 CINTEXLIBDEP           = $(CINTEXLIBDEPM)
 REFEXDICTLIBDEP        = $(REFLEXDICTLIBDEPM)
 RAUTHLIBDEP            = $(RAUTHLIBDEPM)
+GLBSAUTHLIBDEP         = $(GLBSAUTHLIBDEPM)
 KRB5AUTHLIBDEP         = $(KRB5AUTHLIBDEPM)
 SRVAUTHLIBDEP          = $(SRVAUTHLIBDEPM)
 QTGSILIBDEP            = $(QTGSILIBDEPM)
@@ -261,6 +265,7 @@
 SESSIONVIEWERLIBEXTRA   = lib/libProof.lib lib/libGui.lib lib/libHist.lib \
                           lib/libGpad.lib lib/libGraf.lib lib/libTree.lib \
                           lib/libMathCore.lib
+CLARENSLIBEXTRA         = lib/libMathCore.lib
 PEACLIBEXTRA            = lib/libProof.lib lib/libClarens.lib
 PEACGUILIBEXTRA         = lib/libGui.lib
 EGLIBEXTRA              = lib/libGraf3d.lib lib/libGraf.lib lib/libGpad.lib \
@@ -320,6 +325,7 @@
 CINTEXLIBEXTRA          = lib/libReflex.lib
 REFLEXDICTLIBEXTRA      = lib/libReflex.lib
 RAUTHLIBEXTRA           = lib/libNet.lib lib/libRIO.lib
+GLBSAUTHLIBEXTRA        = lib/libRootAuth.lib
 KRB5AUTHLIBEXTRA        = lib/libRootAuth.lib lib/libNet.lib
 SRVAUTHLIBEXTRA         = lib/libNet.lib
 QTGSILIBEXTRA           = lib/libGui.lib lib/libGpad.lib
@@ -389,6 +395,7 @@
 PROOFXLIBEXTRA          = -Llib -lNet -lProof -lThread
 SESSIONVIEWERLIBEXTRA   = -Llib -lProof -lGui -lHist -lGpad -lGraf -lTree \
                           -lMathCore
+CLARENSLIBEXTRA         = -Llib -lMathCore
 PEACLIBEXTRA            = -Llib -lProof -lClarens
 PEACGUILIBEXTRA         = -Llib -lGui
 X3DLIBEXTRA             = -Llib -lGraf3d -lGui
@@ -437,6 +444,7 @@
 CINTEXLIBEXTRA          = -Llib -lReflex
 REFLEXDICTLIBEXTRA      = -Llib -lReflex
 RAUTHLIBEXTRA           = -Llib -lNet -lRIO
+GLBSAUTHLIBEXTRA        = -Llib -lRootAuth
 KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet
 SRVAUTHLIBEXTRA         = -Llib -lNet
 QTGSILIBEXTRA           = -Llib -lGui -lGpad
--- trunk/net/globusauth/Module.mk	2010/07/27 16:09:01	34624
+++ trunk/net/globusauth/Module.mk	2010/07/27 16:18:34	34625
@@ -45,7 +45,7 @@
 include/%.h:    $(GLBSAUTHDIRI)/%.h
 		cp $< $@
 
-$(GLBSAUTHLIB): $(GLBSAUTHO) $(GLBPATCHO) $(ORDER_) $(MAINLIBS)
+$(GLBSAUTHLIB): $(GLBSAUTHO) $(GLBPATCHO) $(ORDER_) $(MAINLIBS) $(GLBSAUTHLIBDEP)
 		@$(MAKELIB) $(PLATFORM) $(LD) "$(LDFLAGS)" \
 		   "$(SOFLAGS)" libGlobusAuth.$(SOEXT) $@ \
 		   "$(GLBSAUTHO) $(GLBPATCHO)" \
