diff -ur root-6.06.02.orig/tutorials/mlp/mlpHiggs.C root-6.06.02/tutorials/mlp/mlpHiggs.C
--- root-6.06.02.orig/tutorials/mlp/mlpHiggs.C	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/tutorials/mlp/mlpHiggs.C	2016-03-23 21:58:56.072799219 +0100
@@ -15,8 +15,8 @@
    // equal to 1 for the signal and 0 for the background is added.
    const char *fname = "mlpHiggs.root";
    TFile *input = 0;
-   if (!gSystem->AccessPathName(fname)) {
-      input = TFile::Open(fname);
+   if (!gSystem->AccessPathName(Form("%s/%s", gSystem->DirName(__FILE__), fname))) {
+      input = TFile::Open(Form("%s/%s", gSystem->DirName(__FILE__), fname));
    } else {
       printf("accessing %s file from http://root.cern.ch/files\n",fname);
       input = TFile::Open(Form("http://root.cern.ch/files/%s",fname));
diff -ur root-6.06.02.orig/tutorials/quadp/portfolio.C root-6.06.02/tutorials/quadp/portfolio.C
--- root-6.06.02.orig/tutorials/quadp/portfolio.C	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/tutorials/quadp/portfolio.C	2016-03-23 22:19:31.309867477 +0100
@@ -273,8 +273,8 @@
 
   const char *fname = "stock.root";
   TFile *f = 0;
-  if (!gSystem->AccessPathName(fname)) {
-     f = TFile::Open(fname);
+  if (!gSystem->AccessPathName(Form("%s/%s", gSystem->DirName(__FILE__), fname))) {
+     f = TFile::Open(Form("%s/%s", gSystem->DirName(__FILE__), fname));
   } else {
      printf("accessing %s file from http://root.cern.ch/files\n",fname);
      f = TFile::Open(Form("http://root.cern.ch/files/%s",fname));
