--- v5-34-00-patches/html/src/THtml.cxx	2012/08/29 09:33:39	45728
+++ v5-34-00-patches/html/src/THtml.cxx	2012/05/08 15:21:21	44178
@@ -310,13 +310,11 @@
          if (!filename.EndsWith(fsentry->GetName()))
             continue;
          fsentry->GetFullName(filesysname, kTRUE); // get the short version
-         filename = filesysname;
          if (!filename.EndsWith(filesysname)) {
-            // It's something - let's see whether we find something better
-            // else leave it as plan B. This helps finding Reflex sources.
-            //filesysname = "";
+            filesysname = "";
             continue;
          }
+         filename = filesysname;
          fsentry->GetFullName(filesysname, kFALSE); // get the long version
          if (fse) *fse = fsentry;
          break;
