diff -ur root-5.34.14.orig/net/davix/src/TDavixSystem.cxx root-5.34.14/net/davix/src/TDavixSystem.cxx
--- root-5.34.14.orig/net/davix/src/TDavixSystem.cxx	2013-12-16 12:19:34.000000000 +0100
+++ root-5.34.14/net/davix/src/TDavixSystem.cxx	2013-12-23 03:47:03.785238174 +0100
@@ -172,7 +172,7 @@
       return 1;
    }
    if (vecRep.size() > 0) {
-      endurl = vecRep[0].getString().c_str();
+      endurl = vecRep[0].uri.getString().c_str();
    } else {
       endurl = path;
    }
