diff -ur root-5.24.00b.orig/configure root-5.24.00b/configure
--- root-5.24.00b.orig/configure	2009-10-11 22:40:04.000000000 +0200
+++ root-5.24.00b/configure	2009-11-26 09:57:20.562269489 +0100
@@ -2916,9 +2916,8 @@
           $cwhich afterimage-config
           # we found afterimage-config - now check the version number
           asimage_version=`afterimage-config --version | awk 'BEGIN { FS="." } { printf "%d", $1 * 1000 + $2 }'`
-          #if test $asimage_version -gt 1003 ; then
-          if test $asimage_version -gt 0 ; then
-             result "WARNING: System libAfterImage is too new, using built-in"
+          if test $asimage_version -lt 1018 ; then
+             result "WARNING: System libAfterImage is too old, using built-in"
           else
              enable_builtin_afterimage="no"
              asextralib=`afterimage-config --libs`
diff -ur root-5.24.00b.orig/graf2d/asimage/src/TASImage.cxx root-5.24.00b/graf2d/asimage/src/TASImage.cxx
--- root-5.24.00b.orig/graf2d/asimage/src/TASImage.cxx	2009-10-11 22:39:57.000000000 +0200
+++ root-5.24.00b/graf2d/asimage/src/TASImage.cxx	2009-11-26 10:26:55.054268550 +0100
@@ -102,6 +102,12 @@
 // auxilary functions for general polygon filling
 #include "TASPolyUtils.c"
 
+// used things from libAfterImage that are not declared in the headers...
+#define ASDrawCTX_CanvasIsARGB	(0x01<<1)
+#define ASDrawCTX_ToolIsARGB	(0x01<<2)
+extern "C" {
+Bool asim_set_custom_brush_colored( ASDrawContext *ctx, ASDrawTool *brush);
+}
 
 ASVisual *TASImage::fgVisual = 0;
 Bool_t TASImage::fgInit = kFALSE;
