diff -ur root-5.34.14.orig/configure root-5.34.14/configure
--- root-5.34.14.orig/configure	2013-12-16 12:19:34.000000000 +0100
+++ root-5.34.14/configure	2013-12-19 03:47:25.973920954 +0100
@@ -4926,10 +4926,14 @@
    if test "x$enable_xrootd" = "xyes"; then
       xrdlibdir="$tmpxrdlibdir"
       if test "x$enable_bonjour" = "xyes"; then
-         check_symbol XrdOucBonjour "$libxrdoucbjr" "$xrdlibdir"
-         if test  $found_symbol -eq 0 ; then
-            bonjourcppflags=
-         fi
+         # This check is broken. The xrootd libraries are C++ hence their
+         # symbol names are mangled. The check_symbol macro is designed
+         # to check for unmangled symbol names in C libraries.
+         # check_symbol XrdOucBonjour "$libxrdoucbjr" "$xrdlibdir"
+         # if test  $found_symbol -eq 0 ; then
+         #    bonjourcppflags=
+         # fi
+         bonjourcppflags=
       fi
    fi
 
