diff -ur root-6.06.02.orig/config/Makefile.freebsd7 root-6.06.02/config/Makefile.freebsd7
--- root-6.06.02.orig/config/Makefile.freebsd7	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.freebsd7	2016-03-20 16:55:12.753322362 +0100
@@ -49,8 +49,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.hurddeb root-6.06.02/config/Makefile.hurddeb
--- root-6.06.02.orig/config/Makefile.hurddeb	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.hurddeb	2016-03-20 16:55:12.753322362 +0100
@@ -42,8 +42,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linux root-6.06.02/config/Makefile.linux
--- root-6.06.02.orig/config/Makefile.linux	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linux	2016-03-20 16:55:12.753322362 +0100
@@ -55,8 +55,10 @@
 else
 F77LIBS      := $(shell $(F77) -m32 -print-file-name=libgfortran.a)
 endif
+ifneq ($(shell $(F77) -m32 -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -m32 -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxalphagcc root-6.06.02/config/Makefile.linuxalphagcc
--- root-6.06.02.orig/config/Makefile.linuxalphagcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxalphagcc	2016-03-20 16:55:12.753322362 +0100
@@ -42,8 +42,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxarm root-6.06.02/config/Makefile.linuxarm
--- root-6.06.02.orig/config/Makefile.linuxarm	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxarm	2016-03-20 16:55:12.753322362 +0100
@@ -49,8 +49,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxarm64 root-6.06.02/config/Makefile.linuxarm64
--- root-6.06.02.orig/config/Makefile.linuxarm64	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxarm64	2016-03-20 16:55:12.753322362 +0100
@@ -53,8 +53,10 @@
 else
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.a)
 endif
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxhppa root-6.06.02/config/Makefile.linuxhppa
--- root-6.06.02.orig/config/Makefile.linuxhppa	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxhppa	2016-03-20 16:55:12.753322362 +0100
@@ -43,8 +43,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxia64gcc root-6.06.02/config/Makefile.linuxia64gcc
--- root-6.06.02.orig/config/Makefile.linuxia64gcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxia64gcc	2016-03-20 16:55:12.753322362 +0100
@@ -42,8 +42,10 @@
 endif
 F77FLAGS      = -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.a)
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxppcgcc root-6.06.02/config/Makefile.linuxppcgcc
--- root-6.06.02.orig/config/Makefile.linuxppcgcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxppcgcc	2016-03-20 16:55:12.753322362 +0100
@@ -50,8 +50,10 @@
 endif
 F77FLAGS      = -m32 -fPIC -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.$(SOEXT))
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a) -lnsl
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxx32gcc root-6.06.02/config/Makefile.linuxx32gcc
--- root-6.06.02.orig/config/Makefile.linuxx32gcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxx32gcc	2016-03-20 16:55:12.753322362 +0100
@@ -54,8 +54,10 @@
 else
 F77LIBS      := $(shell $(F77) -mx32 -print-file-name=libgfortran.a)
 endif
+ifneq ($(shell $(F77) -mx32 -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -mx32 -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.linuxx8664gcc root-6.06.02/config/Makefile.linuxx8664gcc
--- root-6.06.02.orig/config/Makefile.linuxx8664gcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.linuxx8664gcc	2016-03-20 16:55:12.754322372 +0100
@@ -62,8 +62,10 @@
 else
 F77LIBS      := $(shell $(F77) -m64 -print-file-name=libgfortran.a)
 endif
+ifneq ($(shell $(F77) -m64 -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -m64 -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.win32gcc root-6.06.02/config/Makefile.win32gcc
--- root-6.06.02.orig/config/Makefile.win32gcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.win32gcc	2016-03-20 16:55:12.754322372 +0100
@@ -55,8 +55,10 @@
 endif
 F77FLAGS      = -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.a)
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
diff -ur root-6.06.02.orig/config/Makefile.win64gcc root-6.06.02/config/Makefile.win64gcc
--- root-6.06.02.orig/config/Makefile.win64gcc	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/Makefile.win64gcc	2016-03-20 16:55:12.754322372 +0100
@@ -55,8 +55,10 @@
 endif
 F77FLAGS      = -std=legacy
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.a)
+ifneq ($(shell $(F77) -print-file-name=libgfortranbegin.a),libgfortranbegin.a)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 endif
+endif
 
 ifneq ($(findstring g77, $(F77)),)
 ifeq ($(F77),)
