diff -ur root-6.06.02.orig/cmake/modules/SetUpLinux.cmake root-6.06.02/cmake/modules/SetUpLinux.cmake
--- root-6.06.02.orig/cmake/modules/SetUpLinux.cmake	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/cmake/modules/SetUpLinux.cmake	2016-04-06 09:42:10.258085039 +0200
@@ -30,8 +30,8 @@
   else()
     message(FATAL_ERROR "There is no Setup for this compiler up to now. Don't know what to do. Stop cmake at this point.")
   endif()
-elseif(${SYSCTL_OUTPUT} MATCHES ppc64le)
-  message(STATUS "Found a 64bit ppc64 little endian system")
+elseif(${SYSCTL_OUTPUT} MATCHES ppc64)
+  message(STATUS "Found a 64bit PPC system (ppc64/ppc64le)")
   set(SPECIAL_CINT_FLAGS "-DG__64BIT")
   if(CMAKE_COMPILER_IS_GNUCXX)
     message(STATUS "Found GNU compiler collection")
diff -ur root-6.06.02.orig/config/root-config.in root-6.06.02/config/root-config.in
--- root-6.06.02.orig/config/root-config.in	2016-03-03 10:36:03.000000000 +0100
+++ root-6.06.02/config/root-config.in	2016-04-06 09:48:35.837025323 +0200
@@ -205,17 +205,12 @@
    ;;
 linuxppcgcc)
    # PPC Linux with gcc
-   auxcflags="${cxxversionflag} -Wno-deprecated-declarations -fsigned-char -I/usr/X11/include"
+   auxcflags="${cxxversionflag} -Wno-deprecated-declarations -m32 -fsigned-char"
+   auxldflags="-m32"
    auxlibs="-lm -ldl -rdynamic"
    ;;
 linuxppc64gcc)
-   # PPC64 (64 bit mode) Linux with gcc 3.x
-   auxcflags="${cxxversionflag} -Wno-deprecated-declarations -m64 -fsigned-char"
-   auxldflags="-m64"
-   auxlibs="-lm -ldl -rdynamic"
-   ;;
-ppc64le)
-   # PPC64 (64 bit mode) little endian
+   # PPC64/PPC64LE (64 bit mode) Linux with gcc
    auxcflags="${cxxversionflag} -Wno-deprecated-declarations -m64 -fsigned-char"
    auxldflags="-m64"
    auxlibs="-lm -ldl -rdynamic"
