diff -ur root-5.24.00b.orig/hist/hist/src/TGraph2D.cxx root-5.24.00b/hist/hist/src/TGraph2D.cxx
--- root-5.24.00b.orig/hist/hist/src/TGraph2D.cxx	2009-10-11 22:39:45.000000000 +0200
+++ root-5.24.00b/hist/hist/src/TGraph2D.cxx	2009-11-30 23:18:50.892771913 +0100
@@ -155,14 +155,6 @@
 End_Macro
 Begin_Html
 
-2D graphs can be fitted as shown by the following example:
-
-End_Html
-Begin_Macro(source)
-../../../tutorials/fit/graph2dfit.C
-End_Macro
-Begin_Html
-
 Example showing the PCOL option.
 
 End_Html
